<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="tinalasisi" />

<meta name="date" content="2024-11-23" />

<title>Long range probability of a Match</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PODFRIDGE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lasisilab/PODFRIDGE">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Long range probability of a Match</h1>
<h4 class="author">tinalasisi</h4>
<h4 class="date">2024-11-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-12-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PODFRIDGE/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230302code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230302code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230302)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlasisilabPODFRIDGEtreed5b46893df73e807ea6e0017ce7993e9b1d3e6fetargetblankd5b4689a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lasisilab/PODFRIDGE/tree/d5b46893df73e807ea6e0017ce7993e9b1d3e6fe" target="_blank">d5b4689</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlasisilabPODFRIDGEtreed5b46893df73e807ea6e0017ce7993e9b1d3e6fetargetblankd5b4689a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lasisilab/PODFRIDGE/tree/d5b46893df73e807ea6e0017ce7993e9b1d3e6fe" target="_blank">d5b4689</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/racial_proportion_forensic.qmd
    Untracked:  analysis/racial_proportion_report.qmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/long-range-prob.Rmd</code>) and
HTML (<code>docs/long-range-prob.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/228ffbdd8f5dd731e7ef8a1145d7f2dfe754f33c/analysis/long-range-prob.Rmd" target="_blank">228ffbd</a>
</td>
<td>
tinalasisi
</td>
<td>
2024-11-24
</td>
<td>
Add long-range page + fix broken links
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div
id="adapting-the-erlich-et-al.-model-to-incorporate-zero-inflated-negative-binomial-fertility-distributions"
class="section level1">
<h1>Adapting the Erlich et al. Model to Incorporate Zero-Inflated
Negative Binomial Fertility Distributions</h1>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In the Erlich et al. model (2018), the probability of identifying an
individual through genetic genealogy databases is calculated based on
population genetics parameters and assumptions about family sizes. The
original model assumes a Poisson distribution of family sizes with a
constant mean number of children per family and a constant effective
population size. However, real-world fertility data often exhibit
overdispersion and zero-inflation, particularly when considering
different populations.</p>
<p>Our goal is to adapt the original Erlich et al. model to incorporate
a zero-inflated negative binomial (ZINB) distribution for fertility,
using accurate information about the fertility distributions for two
different populations: European Americans and African Americans. We aim
to model the distribution of fertility more accurately while staying as
close as possible to the original model and its assumptions. This
adaptation will allow us to refine the model’s predictions with minimal
changes to the original functions.</p>
<blockquote>
<p><strong>Note:</strong> The ZINB parameters (<code>mu</code>,
<code>theta</code>, <code>pi</code>) need to be estimated accurately
from separate analyses of fertility data for each population. In the
code provided below, these parameters are placeholders and should be
replaced with the actual estimates obtained from your data.</p>
</blockquote>
<hr />
</div>
<div id="original-functions" class="section level2">
<h2>Original Functions</h2>
<p>Here are the original functions from the Erlich et al. model. These
functions calculate the probability of a match for direct and
once-removed relatives and compute the coverage of the database.</p>
<pre class="r"><code># Define genome size and number of chromosomes
genome_size = 35 
num_chrs = 22

# Function to calculate the probability of a match for direct relatives
p_match = function(g, m, min_num_seg) {
  m = m / 100 # Convert m from cM to Morgans
  f = exp(-2 * g * m) / 2^(2 * g - 2) # Calculate f value
  pr = 1 - pbinom(min_num_seg - 1, num_chrs + genome_size * 2 * g, f) # Calculate probability
  return(pr)
}

# Function to calculate the probability of a match for once-removed relatives
p_match_or = function(g, m, min_num_seg) {
  m = m / 100 # Convert m from cM to Morgans
  f = exp(-(2 * g + 1) * m) / 2^(2 * g - 1) # Calculate f value
  pr = 1 - pbinom(min_num_seg - 1, num_chrs + genome_size * (2 * g + 1), f) # Calculate probability
  return(pr)
}

# Function to calculate the coverage of the database
coverage = function(Ks, maxg, N_pop, r, m, min_num_seg, min_num_rel, rep_direct = rep(1, 10), rep_or = rep(1, 10)) {
  N = N_pop / 2 # Convert population size to number of couples
  pr_succ = numeric(length(Ks)) # Initialize the vector of probabilities of success

  # Loop through database sizes
  for (i in seq_along(Ks)) {
    K = Ks[i] # Current database size
    K_same = round(K * (r / 2) / (1 + r / 2)) # Number of direct relatives
    K_or = round(K * 1 / (1 + r / 2)) # Number of once-removed relatives

    # Initialize vectors for probability calculations
    p_no_coal = numeric(maxg) 
    p_coal = numeric(maxg)
    p_no_coal_or = numeric(maxg) 
    p_coal_or = numeric(maxg)
    Ns = N * (r / 2)^(-(1:(maxg + 1))) # Ns values
    tot_p = 0
    tot_p_or = 0

    # Loop through generations
    for (g in 1:maxg) {
      f = 2^(2 * g - 2) / Ns[g] # Coalescence probability for direct relatives
      f_or = 2^(2 * g - 1) / Ns[g + 1] # Coalescence probability for once-removed relatives

      # Update probabilities
      if (g &gt; 1) {
        p_coal[g] = p_no_coal[g - 1] * f
        p_no_coal[g] = p_no_coal[g - 1] * (1 - f)
        p_coal_or[g] = p_no_coal_or[g - 1] * f_or
        p_no_coal_or[g] = p_no_coal_or[g - 1] * (1 - f_or)
      } else {
        p_coal[g] = f 
        p_no_coal[g] = 1 - f 
        p_coal_or[g] = f_or 
        p_no_coal_or[g] = 1 - f_or
      }

      # Update total probabilities
      tot_p = tot_p + p_coal[g] * p_match(g, m, min_num_seg) * rep_direct[g] 
      if (g &lt; maxg) {
        tot_p_or = tot_p_or + p_coal_or[g] * p_match_or(g, m, min_num_seg) * rep_or[g]
      } 
    }

    # Calculate probability of no success
    pr_no_succ = 0
    for (n in 0:(min_num_rel - 1)) {
      for (n_or in 0:n) {
        pr_no_succ = pr_no_succ + dbinom(n_or, K_or, tot_p_or) * dbinom(n - n_or, K_same, tot_p)
      }
    }
    
    # Calculate the probability of success
    pr_succ[i] = 1 - pr_no_succ
  }
  
  return(pr_succ)
}

# Ks: A vector of database sizes
# maxg: Maximum degree of relatedness to consider (e.g., 2 for first cousins)
# N_pop: Total population size
# r: Mean number of children per mating pair
# m: Minimum length in cM of a detectable segment
# min_num_seg: Minimum number of segments required to declare a match
# min_num_rel: Minimum number of relatives required to declare success</code></pre>
<hr />
</div>
<div id="adaptation-of-the-functions" class="section level2">
<h2>Adaptation of the Functions</h2>
<p>We aim to adapt the original functions to incorporate the
zero-inflated negative binomial (ZINB) distribution parameters
(<code>mu</code>, <code>theta</code>, <code>pi</code>) that accurately
represent the fertility distributions for European Americans and African
Americans. This involves modifying the coverage function to adjust the
coalescence probability based on the variance in family sizes due to the
ZINB distribution.</p>
<blockquote>
<p><strong>Note:</strong> The ZINB parameters (<code>mu</code>,
<code>theta</code>, <code>pi</code>) need to be estimated accurately
from separate analyses of fertility data for each population. In the
code below, these parameters are placeholders and should be replaced
with your actual estimates.</p>
</blockquote>
<hr />
</div>
<div id="adapted-functions" class="section level2">
<h2>Adapted Functions</h2>
<pre class="r"><code># Define genome size and number of chromosomes
genome_size = 35 
num_chrs = 22

# Function to calculate the probability of a match for direct relatives (unchanged)
p_match = function(g, m, min_num_seg) {
  m = m / 100 # Convert m from cM to Morgans
  f = exp(-2 * g * m) / 2^(2 * g - 2) # Calculate f value
  pr = 1 - pbinom(min_num_seg - 1, num_chrs + genome_size * 2 * g, f) # Calculate probability
  return(pr)
}

# Function to calculate the probability of a match for once-removed relatives (unchanged)
p_match_or = function(g, m, min_num_seg) {
  m = m / 100 # Convert m from cM to Morgans
  f = exp(-(2 * g + 1) * m) / 2^(2 * g - 1) # Calculate f value
  pr = 1 - pbinom(min_num_seg - 1, num_chrs + genome_size * (2 * g + 1), f) # Calculate probability
  return(pr)
}

# Adapted coverage function to include ZINB parameters
coverage = function(Ks, maxg, N_pop, m, min_num_seg, min_num_rel, mu, theta, pi) {
  N = N_pop / 2 # Convert population size to number of couples
  pr_succ = numeric(length(Ks)) # Initialize the vector of probabilities of success

  # Calculate variance from ZINB
  variance = (1 - pi) * mu * (1 + mu * (1 / theta + pi))
  
  # Adjusted coalescence probability factor due to variance in family size
  sigma2_mu2 = variance / mu^2
  adjustment_factor = (1 + sigma2_mu2) # Adjustment factor for coalescence probability

  # Loop through database sizes
  for (i in seq_along(Ks)) {
    K = Ks[i] # Current database size
    K_same = round(K / 2) # Assuming half of the database are direct relatives
    K_or = K - K_same

    # Initialize vectors for probability calculations
    p_no_coal = numeric(maxg) 
    p_coal = numeric(maxg)
    p_no_coal_or = numeric(maxg) 
    p_coal_or = numeric(maxg)
    tot_p = 0
    tot_p_or = 0

    # Loop through generations
    for (g in 1:maxg) {
      # Adjusted coalescence probabilities incorporating variance
      f = (2^(2 * g - 2) / N) * adjustment_factor # Coalescence probability for direct relatives
      f_or = (2^(2 * g - 1) / N) * adjustment_factor # Coalescence probability for once-removed relatives

      # Update probabilities
      if (g &gt; 1) {
        p_coal[g] = p_no_coal[g - 1] * f
        p_no_coal[g] = p_no_coal[g - 1] * (1 - f)
        p_coal_or[g] = p_no_coal_or[g - 1] * f_or
        p_no_coal_or[g] = p_no_coal_or[g - 1] * (1 - f_or)
      } else {
        p_coal[g] = f 
        p_no_coal[g] = 1 - f 
        p_coal_or[g] = f_or 
        p_no_coal_or[g] = 1 - f_or
      }

      # Update total probabilities
      tot_p = tot_p + p_coal[g] * p_match(g, m, min_num_seg)
      if (g &lt; maxg) {
        tot_p_or = tot_p_or + p_coal_or[g] * p_match_or(g, m, min_num_seg)
      } 
    }

    # Calculate probability of no success
    pr_no_succ = 0
    for (n in 0:(min_num_rel - 1)) {
      for (n_or in 0:n) {
        pr_no_succ = pr_no_succ + dbinom(n_or, K_or, tot_p_or) * dbinom(n - n_or, K_same, tot_p)
      }
    }
    
    # Calculate the probability of success
    pr_succ[i] = 1 - pr_no_succ
  }
  
  return(pr_succ)
}</code></pre>
<hr />
</div>
<div id="application-of-the-adapted-functions" class="section level2">
<h2>Application of the Adapted Functions</h2>
<pre class="r"><code># Population sizes
EA_POP_1990 = 199686070 # European American Population 1990
AA_POP_1990 = 29986060  # African American Population 1990

# Placeholder ZINB parameters for European Americans (replace with actual estimates)
N_pop_white = EA_POP_1990
mu_white = 2.79    # Mean number of children among women with at least one child
theta_white = 1.25 # Dispersion parameter from ZINB model
pi_white = 0.197   # Zero-inflation probability

# Placeholder ZINB parameters for African Americans (replace with actual estimates)
N_pop_black = AA_POP_1990
mu_black = 3.06    # Mean number of children among women with at least one child
theta_black = 0.95 # Dispersion parameter from ZINB model
pi_black = 0.179   # Zero-inflation probability

# Common parameters
num_K = 10000      # Number of data points
m = 6              # Minimum cM
min_num_seg = 2    # Number of segments
min_num_rel = 1    # Minimum number of relatives
maxg_values = c(2, 3, 4, 5) # Max generations for 1C to 4C

# Generate Ks (database sizes) for both populations
Ks_white = round(seq(from = N_pop_white / num_K, to = N_pop_white, length.out = num_K))
Ks_black = round(seq(from = N_pop_black / num_K, to = N_pop_black, length.out = num_K))

# Initialize lists to store coverage results
coverage_white_list = list()
coverage_black_list = list()

# Calculate coverage for each cousin degree for European Americans
for (maxg in maxg_values) {
  coverage_white_list[[as.character(maxg)]] = coverage(
    Ks = Ks_white, maxg = maxg, N_pop = N_pop_white, m = m,
    min_num_seg = min_num_seg, min_num_rel = min_num_rel,
    mu = mu_white, theta = theta_white, pi = pi_white
  )
}

# Calculate coverage for each cousin degree for African Americans
for (maxg in maxg_values) {
  coverage_black_list[[as.character(maxg)]] = coverage(
    Ks = Ks_black, maxg = maxg, N_pop = N_pop_black, m = m,
    min_num_seg = min_num_seg, min_num_rel = min_num_rel,
    mu = mu_black, theta = theta_black, pi = pi_black
  )
}</code></pre>
<hr />
</div>
<div id="plotting-the-results" class="section level2">
<h2>Plotting the Results</h2>
<pre class="r"><code>library(ggplot2)
library(RColorBrewer)
library(dplyr)

# Prepare data for plotting

# For European Americans
plot_data_white = data.frame(
  Ks_proportion = rep(Ks_white / N_pop_white * 100, times = length(maxg_values)),
  Coverage = unlist(coverage_white_list),
  Population = &quot;European Americans&quot;,
  Relationship = factor(rep(c(&quot;1C&quot;, &quot;2C&quot;, &quot;3C&quot;, &quot;4C&quot;), each = num_K), levels = c(&quot;1C&quot;, &quot;2C&quot;, &quot;3C&quot;, &quot;4C&quot;))
)

# For African Americans
plot_data_black = data.frame(
  Ks_proportion = rep(Ks_black / N_pop_black * 100, times = length(maxg_values)),
  Coverage = unlist(coverage_black_list),
  Population = &quot;African Americans&quot;,
  Relationship = factor(rep(c(&quot;1C&quot;, &quot;2C&quot;, &quot;3C&quot;, &quot;4C&quot;), each = num_K), levels = c(&quot;1C&quot;, &quot;2C&quot;, &quot;3C&quot;, &quot;4C&quot;))
)

# Combine data
plot_data = rbind(plot_data_white, plot_data_black)

# Remove rows with NA or infinite Coverage values
plot_data &lt;- plot_data[is.finite(plot_data$Coverage), ]

# Create color palette for cousin degrees
colors = brewer.pal(4, &quot;Set1&quot;)  # Use a palette with 4 distinct colors

# Create the plot
p &lt;- ggplot(plot_data, aes(x = Ks_proportion, y = Coverage, color = Relationship)) +
  geom_line() +
  scale_color_manual(values = colors, name = &quot;Cousin Degree&quot;) +
  scale_x_continuous(labels = scales::percent_format(scale = 1), breaks = seq(0, 5, 1), limits = c(0, 5)) +
  labs(title = &quot;Coverage Comparison Between European and African Americans&quot;,
       x = &quot;Database Size (% of Population)&quot;,
       y = &quot;Probability of a Match&quot;) +
  theme_minimal() +
  facet_wrap(~Population, ncol = 2) +
  theme(legend.position = &quot;bottom&quot;)

# Add labels directly on the lines
label_data &lt;- plot_data %&gt;%
  group_by(Population, Relationship) %&gt;%
  filter(Ks_proportion == max(Ks_proportion))

# Adjust the label positions slightly
label_data$Ks_proportion &lt;- label_data$Ks_proportion + 0.1  # Shift labels slightly to the right

p &lt;- p + geom_text(data = label_data,
                   aes(label = Relationship),
                   hjust = 0,
                   vjust = 0,
                   size = 3,
                   show.legend = FALSE)

# Adjust x limits to make space for labels
p &lt;- p + xlim(0, 5.5)

# Display the plot
print(p)</code></pre>
<hr />
</div>
<div id="conclusion" class="section level2">
<h2>Conclusion</h2>
<p>By adapting the original functions to incorporate the ZINB
parameters, we have refined the model to more accurately represent the
fertility distributions of European Americans and African Americans. We
have made minimal changes to the original model, staying as close as
possible to its assumptions and structure. This approach allows us to
enhance the model’s accuracy with minimal effort, making it accessible
for further analysis and contributions.</p>
<hr />
</div>
<div id="references" class="section level2">
<h2>References</h2>
<ul>
<li>Erlich, Y., Shor, T., Pe’er, I., &amp; Carmi, S. (2018). Identity
inference of genomic data using long-range familial searches.
<em>Science</em>, 362(6415), 690–694.</li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=Chinese (Simplified)_China.utf8 
[2] LC_CTYPE=Chinese (Simplified)_China.utf8   
[3] LC_MONETARY=Chinese (Simplified)_China.utf8
[4] LC_NUMERIC=C                               
[5] LC_TIME=Chinese (Simplified)_China.utf8    

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] vctrs_0.6.4       httr_1.4.7        cli_3.6.1         knitr_1.44       
 [5] rlang_1.1.1       xfun_0.40         stringi_1.7.12    processx_3.8.2   
 [9] promises_1.2.1    jsonlite_1.8.8    glue_1.6.2        rprojroot_2.0.4  
[13] git2r_0.35.0      htmltools_0.5.6.1 httpuv_1.6.11     ps_1.7.5         
[17] sass_0.4.7        fansi_1.0.5       rmarkdown_2.25    jquerylib_0.1.4  
[21] tibble_3.2.1      evaluate_0.23     fastmap_1.1.1     yaml_2.3.7       
[25] lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.0     compiler_4.3.1   
[29] fs_1.6.3          pkgconfig_2.0.3   Rcpp_1.0.11       rstudioapi_0.15.0
[33] later_1.3.1       digest_0.6.33     R6_2.5.1          utf8_1.2.3       
[37] pillar_1.9.0      callr_3.7.5       magrittr_2.0.3    bslib_0.5.1      
[41] tools_4.3.1       cachem_1.0.8      getPass_0.2-4    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
