<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Junhui He" />


<title>Disparities in the probability of finding a relative via long-range familial search</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PODFRIDGE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lasisilab/PODFRIDGE">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Disparities in the probability of finding a
relative via long-range familial search</h1>
<h4 class="author">Junhui He</h4>
<h4 class="date">2025-01-22 20:24:56</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PODFRIDGE/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230302code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230302code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230302)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlasisilabPODFRIDGEtree90a3d7c710bd9e1ee2853d1dd36c0a18976919aftargetblank90a3d7ca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lasisilab/PODFRIDGE/tree/90a3d7c710bd9e1ee2853d1dd36c0a18976919af" target="_blank">90a3d7c</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlasisilabPODFRIDGEtree90a3d7c710bd9e1ee2853d1dd36c0a18976919aftargetblank90a3d7ca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lasisilab/PODFRIDGE/tree/90a3d7c710bd9e1ee2853d1dd36c0a18976919af" target="_blank">90a3d7c</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  data/family_size.csv

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitattributes
    Modified:   .gitignore
    Modified:   PODFRIDGE.Rproj
    Modified:   README.md
    Modified:   _workflowr.yml
    Modified:   analysis/CODIS-STR-freqs.Rmd
    Modified:   analysis/STR-simulation.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/database-composition.Rmd
    Modified:   analysis/final_equation.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd
    Modified:   analysis/long-range-prob.Rmd
    Modified:   analysis/lr.Rmd
    Modified:   analysis/lr_dist.Rmd
    Modified:   analysis/methods.Rmd
    Modified:   analysis/murphy.rmd
    Modified:   analysis/regression.Rmd
    Modified:   analysis/relative-distribution.Rmd
    Modified:   code/README.md
    Modified:   code/STR-sims-env.txt
    Modified:   code/STR_sims-v2.R
    Modified:   code/STR_sims-v3.R
    Modified:   code/STR_sims.R
    Modified:   code/assemble_table.R
    Modified:   code/env.yml
    Modified:   code/geno-sim_script.sh
    Modified:   code/great-lakes-setup.sh
    Modified:   code/known-vs-tested_simulation_script.R
    Modified:   code/launcher_job
    Modified:   code/lr.R
    Modified:   code/lr_cutoff.R
    Modified:   code/lr_expanded.R
    Modified:   code/lr_plots.R
    Modified:   code/rslurm_simulate_genotypes.R
    Modified:   code/script-v2.sh
    Modified:   code/script-v3-array1.sh
    Modified:   code/script-v3-batch1.sh
    Modified:   code/script.sh
    Modified:   code/script_v3-launcher.sh
    Modified:   code/simulate_focal_genotypes.R
    Modified:   code/simulate_genotypes.R
    Modified:   code/simulate_genotypes_expanded.R
    Modified:   code/simulations_script.R
    Modified:   code/synthetic_data.R
    Modified:   data.csv
    Modified:   data/1036_AfAm.csv
    Modified:   data/1036_AfAm_wide.csv
    Modified:   data/1036_Asian.csv
    Modified:   data/1036_Asian_wide.csv
    Modified:   data/1036_Cauc.csv
    Modified:   data/1036_Cauc_wide.csv
    Modified:   data/1036_Hispanic.csv
    Modified:   data/1036_Hispanic_wide.csv
    Modified:   data/1036_all.csv
    Modified:   data/1036_all_wide.csv
    Modified:   data/Coop_cousins_vs_database_size.R
    Modified:   data/DTC_access_export.csv
    Modified:   data/DTC_race_export.csv
    Modified:   data/Murphy_Appendix_all_data.csv
    Modified:   data/README.md
    Modified:   data/autosomal_ibd_by_pair.20150206.txt
    Modified:   data/data_filtered_recoded.csv
    Modified:   data/data_proportions.csv
    Modified:   data/df_allelefreq_combined.csv
    Modified:   data/df_allelefreq_wide.csv
    Modified:   data/dl_known_vs_tested_simulation_results.csv
    Modified:   data/dtc_demographics.csv
    Modified:   data/erlich2018.md
    Modified:   data/final_CODIS_data.csv
    Modified:   data/frequency_table_by_race_year.csv
    Modified:   data/marker_list.txt
    Modified:   data/murphy_foia_cleaned.csv
    Modified:   data/populations_states.csv
    Modified:   data/processed_genotypes.csv
    Modified:   data/proportions_table_by_race_year.csv
    Modified:   data/regression_data/readme
    Modified:   data/sim-summary_genotypes.csv
    Modified:   data/sim_processed_genotypes.csv
    Modified:   data/sim_summary_genotypes.csv
    Modified:   data/sims/simulation_11297218/processed_genotypes.csv
    Modified:   data/sims/simulation_11297218/timing_log.csv
    Modified:   data/simulation_results.csv
    Modified:   data/stats_by_race_year.csv
    Modified:   data/subset_example_data/readme
    Modified:   data/subset_example_data/sim_cutoffs.csv
    Modified:   data/subset_example_data/sim_proportions_exceeding_cutoffs.csv
    Modified:   data/summary_genotypes.csv
    Modified:   output/README.md
    Modified:   output/aggregated_llr_results_by_population.csv
    Modified:   output/cutoff_results.csv
    Modified:   output/cutoffs.csv
    Modified:   output/demographic_composition_comparison.csv
    Modified:   output/proportions_exceeding_cutoffs.csv
    Modified:   output/sim_cutoffs.csv
    Modified:   output/sim_proportions_exceeding_cutoffs.csv
    Modified:   output/simulation_11297164/sim_processed_genotypes.csv
    Modified:   output/simulation_20240726-103909/sim_cutoffs.csv
    Modified:   output/simulation_20240726-103909/sim_processed_genotypes.csv
    Modified:   output/simulation_20240726-103909/sim_proportions_exceeding_cutoffs.csv
    Modified:   output/simulation_20240726-103909/sim_summary_genotypes.csv
    Modified:   output/simulation_20240726-103909/timing_log.csv
    Modified:   output/simulation_20240726-152621/sim_cutoffs.csv
    Modified:   output/simulation_20240726-152621/sim_processed_genotypes.csv
    Modified:   output/simulation_20240726-152621/sim_proportions_exceeding_cutoffs.csv
    Modified:   output/simulation_20240726-152621/sim_summary_genotypes.csv
    Modified:   output/simulation_20240726-152621/timing_log.csv
    Modified:   output/simulation_20240726-154225/logfiles/resource_usage_11226865.log
    Modified:   output/simulation_20240726-154225/logfiles/usage_11226865.log
    Modified:   output/simulation_20240726-154225/sim_cutoffs.csv
    Modified:   output/simulation_20240726-154225/sim_processed_genotypes.csv
    Modified:   output/simulation_20240726-154225/sim_proportions_exceeding_cutoffs.csv
    Modified:   output/simulation_20240726-154225/sim_summary_genotypes.csv
    Modified:   output/simulation_20240726-154225/timing_log.csv
    Modified:   output/simulation_20240726-155743/sim_cutoffs.csv
    Modified:   output/simulation_20240726-155743/sim_processed_genotypes.csv
    Modified:   output/simulation_20240726-155743/sim_proportions_exceeding_cutoffs.csv
    Modified:   output/simulation_20240726-155743/sim_summary_genotypes.csv
    Modified:   output/simulation_20240726-155743/slurm/resource_usage_11226959.log
    Modified:   output/simulation_20240726-155743/slurm/usage_11226959.log
    Modified:   output/simulation_20240726-155743/timing_log.csv
    Modified:   output/simulation_20240726-160726_11227723/sim_cutoffs.csv
    Modified:   output/simulation_20240726-160726_11227723/sim_processed_genotypes.csv
    Modified:   output/simulation_20240726-160726_11227723/sim_proportions_exceeding_cutoffs.csv
    Modified:   output/simulation_20240726-160726_11227723/sim_summary_genotypes.csv
    Modified:   output/simulation_20240726-160726_11227723/slurm/resource_usage_11227723.log
    Modified:   output/simulation_20240726-160726_11227723/slurm/usage_11227723.log
    Modified:   output/simulation_20240726-160726_11227723/timing_log.csv
    Modified:   output/simulation_20240726-162034_11228488/slurm/resource_usage_11228488.log
    Modified:   output/simulation_20240726-162034_11228488/slurm/usage_11228488.log
    Modified:   output/simulation_20240726-163235_11228791/slurm/resource_usage_11228791.log
    Modified:   output/simulation_20240726-163235_11228791/slurm/usage_11228791.log
    Modified:   output/simulation_20240726-164223_11229998/slurm/resource_usage_11229998.log
    Modified:   output/simulation_20240726-164223_11229998/slurm/usage_11229998.log
    Modified:   output/simulation_20240726-165046_11230240/slurm/resource_usage_11230240.log
    Modified:   output/simulation_20240726-165046_11230240/slurm/usage_11230240.log
    Modified:   output/simulation_20240726-165603/sim_cutoffs.csv
    Modified:   output/simulation_20240726-165603/sim_processed_genotypes.csv
    Modified:   output/simulation_20240726-165603/sim_proportions_exceeding_cutoffs.csv
    Modified:   output/simulation_20240726-165603/sim_summary_genotypes.csv
    Modified:   output/simulation_20240726-165603/timing_log.csv
    Modified:   output/simulation_20240726-165924/sim_cutoffs.csv
    Modified:   output/simulation_20240726-165924/sim_processed_genotypes.csv
    Modified:   output/simulation_20240726-165924/sim_proportions_exceeding_cutoffs.csv
    Modified:   output/simulation_20240726-165924/sim_summary_genotypes.csv
    Modified:   output/simulation_20240726-165924/timing_log.csv
    Modified:   output/simulation_20240726-170852/sim_cutoffs.csv
    Modified:   output/simulation_20240726-170852/sim_processed_genotypes.csv
    Modified:   output/simulation_20240726-170852/sim_proportions_exceeding_cutoffs.csv
    Modified:   output/simulation_20240726-170852/sim_summary_genotypes.csv
    Modified:   output/simulation_20240726-170852/timing_log.csv
    Modified:   output/simulation_20240726-172213/sim_cutoffs.csv
    Modified:   output/simulation_20240726-172213/sim_processed_genotypes.csv
    Modified:   output/simulation_20240726-172213/sim_proportions_exceeding_cutoffs.csv
    Modified:   output/simulation_20240726-172213/sim_summary_genotypes.csv
    Modified:   output/simulation_20240726-172230/sim_cutoffs.csv
    Modified:   output/simulation_20240726-172230/sim_processed_genotypes.csv
    Modified:   output/simulation_20240726-172230/sim_proportions_exceeding_cutoffs.csv
    Modified:   output/simulation_20240726-172230/sim_summary_genotypes.csv
    Modified:   output/simulation_20240726-172230/timing_log.csv
    Modified:   output/simulation_20240727-143100/sim_cutoffs.csv
    Modified:   output/simulation_20240727-143100/sim_processed_genotypes.csv
    Modified:   output/simulation_20240727-143100/sim_proportions_exceeding_cutoffs.csv
    Modified:   output/simulation_20240727-143100/sim_summary_genotypes.csv
    Modified:   slurm/STR_sims-11266147.log
    Modified:   slurm/STR_sims-11297218.log
    Modified:   slurm/STR_sims-v2-11266148.log
    Modified:   slurm/STR_sims-v2-11297164.log
    Modified:   slurm/resource_usage_11226557.log
    Modified:   slurm/resource_usage_11231120.log
    Modified:   slurm/resource_usage_11233712.log
    Modified:   slurm/resource_usage_11234141.log
    Modified:   slurm/resource_usage_11234476.log
    Modified:   slurm/resource_usage_11234482.log
    Modified:   slurm/resource_usage_11266147.log
    Modified:   slurm/resource_usage_11266148.log
    Modified:   slurm/resource_usage_11297164.log
    Modified:   slurm/usage_11226557.log
    Modified:   slurm/usage_11231120.log
    Modified:   slurm/usage_11233712.log
    Modified:   slurm/usage_11234141.log
    Modified:   slurm/usage_11234476.log
    Modified:   slurm/usage_11234482.log
    Modified:   slurm/usage_11266147.log
    Modified:   slurm/usage_11266148.log
    Modified:   slurm/usage_11297164.log
    Modified:   slurm/usage_11297218.log

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/probability_disparity.Rmd</code>)
and HTML (<code>docs/probability_disparity.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/90a3d7c710bd9e1ee2853d1dd36c0a18976919af/analysis/probability_disparity.Rmd" target="_blank">90a3d7c</a>
</td>
<td>
Junhui He
</td>
<td>
2025-01-22
</td>
<td>
wflow_publish("analysis/probability_disparity.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/8c86890ef53a87c172aaf94be7782a828f89dd37/docs/probability_disparity.html" target="_blank">8c86890</a>
</td>
<td>
Junhui He
</td>
<td>
2025-01-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/64c4f90fbc623e0c1baeb950e36965e1abe1ae3e/analysis/probability_disparity.Rmd" target="_blank">64c4f90</a>
</td>
<td>
Junhui He
</td>
<td>
2025-01-17
</td>
<td>
create a report for Result 3
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objective" class="section level2">
<h2>1 Objective</h2>
<p>We aims to determine the difference in the probability of finding a
match in a direct-to-consumer (DTC) genetic database for Black and White
Americans. This analysis integrates family size distributions (from
Result 2), database representation disparities (from Result 1B), and
considers the proportion of DTC databases accessible to law
enforcement.</p>
<p>We establish a theoretical binomial model <span
class="math inline">\(\text{Bin}(K,p)\)</span> to calculate the
probability of finding a match in a database, where <span
class="math inline">\(K\)</span> is the size of the database and <span
class="math inline">\(p\)</span> is the match probability between a
target and another individual. The match probability <span
class="math inline">\(p\)</span> is calculated based on the population
size, the database size, racial representation disparities, and the
family size distribution.</p>
</div>
<div id="model-assumptions" class="section level2">
<h2>2 Model assumptions</h2>
<ol style="list-style-type: decimal">
<li><p>In the current generation, the population size is <span
class="math inline">\(N\)</span>, and the racial proportion of the
population is <span class="math inline">\(\alpha\)</span>.</p></li>
<li><p>The database has <span class="math inline">\(K\)</span>
individuals, and the racial proportion of the database is <span
class="math inline">\(\beta\)</span>. The database is randomly sampled
from the current population. Both the population proportion <span
class="math inline">\(\alpha\)</span> and the database proportion <span
class="math inline">\(\beta\)</span> are considered fixed
constants.</p></li>
<li><p>The family trees of different races are strictly separated. For
instance, the ancestors and descendants of black Americans are also
black Americans.</p></li>
<li><p>We assume that there is no inter-marriage in the family involved,
except for the known ancestor couple of the target.</p></li>
<li><p>The number of children per couple at the generation <span
class="math inline">\(g\)</span> before the present is <span
class="math inline">\(r_g\)</span>. It is assumed that <span
class="math inline">\(r_g\)</span> follows a multinomial distribution.
In this model, the distribution of <span
class="math inline">\(r_1\)</span> is estimated based on the number of
children born to women aged 40-49 in 1990, the distribution of <span
class="math inline">\(r_2\)</span> is estimated based on the number of
children born to women aged 40-49 in 1960, and the distribution of <span
class="math inline">\(r_3\)</span> is estimated based on the number of
children born to women aged 70+ in 1960. For generations <span
class="math inline">\(g\geq 3\)</span>, <span
class="math inline">\(r_g\)</span> is assumed to follow the same
distribution as <span class="math inline">\(r_3\)</span>. Additionally,
if a specific couple is known as the ancestors of the target, it is
assumed that they have at least one child.</p></li>
<li><p>Individuals are diploid and we consider only the autosomal
genome.</p></li>
<li><p>The genome of the target individual is compared to those of all
individuals in the database, and identical-by-descent (IBD) segments are
identified. We assume that detectable segments must be of length <span
class="math inline">\(\geq\)</span> <span
class="math inline">\(m\)</span> (in Morgans). We further assume that in
order to confidently detect the relationship (a “match”), we must
observe at least <span class="math inline">\(s\)</span> such
segments</p></li>
<li><p>We only consider relationships for which the common ancestors
have lived <span class="math inline">\(g\leq g_{\max}\)</span>
generations ago. For example, <span class="math inline">\(g=1\)</span>
for siblings, <span class="math inline">\(g=2\)</span> for first
cousins, etc. All cousins/siblings are full.</p></li>
<li><p>We only consider regular cousins, excluding once removed and so
on.</p></li>
<li><p>The number of matches between the target and the individuals in
the database is counted. If we have more than <span
class="math inline">\(t\)</span> matches, we declare that there is
sufficient information to trace back the target. Typically, we simply
assume <span class="math inline">\(t=1\)</span>.</p></li>
</ol>
</div>
<div id="derivation" class="section level2">
<h2>3 Derivation</h2>
<div id="the-probability-of-a-sharing-a-pair-of-ancestor"
class="section level3">
<h3>3.1 The probability of a sharing a pair of ancestor</h3>
<p>Consider the cousins of the target. <span
class="math inline">\(g\)</span> generations before the present, the
target has <span class="math inline">\(2^{g-1}\)</span> ancestral
couples. For example, each individual has one pair of parents (<span
class="math inline">\(g=1\)</span>), two pairs of grandparents (<span
class="math inline">\(g=2\)</span>), four pairs of great-grandparents
(<span class="math inline">\(g=3\)</span>) and so on. Each ancestral
couple contributes to <span class="math inline">\((r_g-1)
\prod_{i=1}^{g-1} r_i\)</span> of the <span
class="math inline">\((g-1)\)</span>-th cousins of the target. For
example, consider a pair of grandparents (<span
class="math inline">\(g=2\)</span>), each individual has <span
class="math inline">\((r_2-1)\)</span> uncles/aunts, and hence <span
class="math inline">\((r_2-1)r_1\)</span> first cousins. Note that <span
class="math inline">\(r_g \geq 1\)</span>. Therefore, the total number
of the <span class="math inline">\((g-1)\)</span>-th cousins is given
by</p>
<p><span class="math display">\[\text{The number of the $(g-1)$
cousins}=2^{g-1}(r_g-1) \prod_{i=1}^{g-1} r_i.\]</span></p>
<p>Under the assumption of separated family trees and randomly sampled
database, given the family size, the probability to share an ancestral
couple for the first time at generation <span
class="math inline">\(g\)</span> between the target and the individual
in the database with the same race is approximately:</p>
<p><span class="math display">\[P(\text{first sharing a mating pair at
$g$ for a certain race}|r)=\frac{2^{g-1}(r_g-1) \prod_{i=1}^{g-1}
r_i}{\alpha N}.\]</span></p>
</div>
<div id="the-probability-of-a-match-given-a-shared-mating-pair"
class="section level3">
<h3>3.2 The probability of a match given a shared mating pair</h3>
<p>This probability is irrelevant with the race. We adopt the estimation
in the paper of Erlich et al. 2018. To declare a match, we need at least
<span class="math inline">\(s\)</span> segments of at least length <span
class="math inline">\(m\)</span>. Thus, given a first shared mating pair
<span class="math inline">\(g\)</span> generations ago, the probability
to observe a match is</p>
<p><span class="math display">\[P(\text{match}|g)=1-\sum_{k=0}^{s-1}
\text{Bin}(k;2Lg+22,\frac{\exp(-2mg)}{2^{2g-2}}),\]</span> where <span
class="math inline">\(L\)</span> is the total genome length, roughly
<span class="math inline">\(L=35\)</span>.</p>
</div>
<div
id="the-match-probability-between-a-target-and-a-individual-from-the-same-race"
class="section level3">
<h3>3.3 The match probability between a target and a individual from the
same race</h3>
<p>Given the family size, the probability of declaring a match between
the target and a random individual in the database is simply the sum of
the product over all <span class="math inline">\(g\)</span>,</p>
<p><span class="math display">\[ P(\text{match}|r)=\sum_{g=1}^{g_{\max}}
P(\text{match}|g) \frac{2^{g-1}(r_g-1) \prod_{i=1}^{g-1} r_i}{\alpha N}.
\]</span></p>
<p>The number of matches to a database is assumed to follow a binomial
distribution defined as</p>
<p><span class="math display">\[ \text{Bin}(\beta K, P(\text{match}|r))
.\]</span></p>
<p>To identify an individual, we need to find at least <span
class="math inline">\(t\)</span> matches in the database. Thus, given
the family size,</p>
<p><span class="math display">\[P(\text{identify}|r)=1-\sum_{k=1}^{t-1}
\text{Bin}(k;\beta K, P(\text{match}|r)).\]</span></p>
<p>We utilize Monte Carlo methods to calculate the mean probability of
identifying an individual over family size, i.e., <span
class="math inline">\(E[P(\text{identify}|r)]\)</span>.</p>
<p>Additionally, since <span class="math inline">\(r\)</span> is a
random variable, we calculate <span
class="math inline">\(P(\text{identify}|\bar{r})\)</span> to make a
comparison with <span
class="math inline">\(E[P(\text{identify}|r)]\)</span>, where <span
class="math inline">\(\bar{r}\)</span> represents the mean number of
children. This probability <span
class="math inline">\(P(\text{identify}|\bar{r})\)</span> is used in the
paper of Erlich et al. 2018, if we substitute <span
class="math inline">\(\bar{r}\)</span> by a constant <span
class="math inline">\(r\)</span>.</p>
</div>
</div>
<div id="experimental-results" class="section level2">
<h2>4 Experimental results</h2>
<div id="identifying-probabilities-considering-the-family-size"
class="section level3">
<h3>4.1 Identifying probabilities considering the family size</h3>
<p>In this subsection, it is assumed that the population and database
consist of a single race. Ratios <span
class="math inline">\(\alpha\)</span> and <span
class="math inline">\(\beta\)</span> are set to <span
class="math inline">\(1\)</span> in the calculation of identifying
probabilities. These probabilities are utilized to explore the influence
of disparities in the family size of different races, without
considering the racial representation.</p>
<ol style="list-style-type: decimal">
<li>The mean probability of identifying an individual over the family
size up to a certain type of cousinship from a pure database, <span
class="math inline">\(E[P(\text{identify}|r)]\)</span></li>
</ol>
<p><img src="figure/probability_disparity.Rmd/plot-probability-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<ol start="2" style="list-style-type: decimal">
<li>The pseudo probability of identifying an individual using a mean
family size up to a certain type of cousinship from a pure database,
<span class="math inline">\(P(\text{identify}|\bar{r})\)</span></li>
</ol>
<p><img src="figure/probability_disparity.Rmd/plot-pseudo-probability-1-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="identifying-probabilities-considering-the-family-size-and-racial-representation"
class="section level3">
<h3>4.2 Identifying probabilities considering the family size and racial
representation</h3>
<p>In this subsection, it is assumed that the population and database
consist of multiple races, including black and white Americans. Ratios
<span class="math inline">\(\alpha\)</span> and <span
class="math inline">\(\beta\)</span> correspond to racial proportions in
population and database, respectively. We calculate the probabilities of
identifying an individual by treating <span
class="math inline">\(\alpha\)</span> and <span
class="math inline">\(\beta\)</span> fixed as constants. When the
database size is much smaller than the population size, this assumption
seems to be reasonable.</p>
<ol style="list-style-type: decimal">
<li>The mean probability of identifying an individual over the family
size up to a certain type of cousinship from a mixed database, <span
class="math inline">\(E[P(\text{identify}|r)]\)</span>, considering the
population size, the database size and the race</li>
</ol>
<p><img src="figure/probability_disparity.Rmd/plot-probability-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<ol start="2" style="list-style-type: decimal">
<li>The pseudo probability of identifying an individual using a mean
family size up to a certain type of cousinship from a mixed database,
<span class="math inline">\(P(\text{identify}|\bar{r})\)</span>,
considering the population size, the database size and the race</li>
</ol>
<p><img src="figure/probability_disparity.Rmd/plot-pseudo-probability-2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="conclusions" class="section level2">
<h2>5 Conclusions</h2>
<ol style="list-style-type: decimal">
<li><p>Considering only the family size, it is interesting to observe
that identification probabilities <span
class="math inline">\(E[P(\text{identify}|r)]\)</span> for black and
white Americans are close to each other (see Figure 1 in Section 4.1),
while pseudo identification probabilities <span
class="math inline">\(P(\text{identify}|\bar{r})\)</span> for black
Americans are significantly higher than that for black Americans (see
Figure 2 in Section 4.1).</p></li>
<li><p>Considering the family size and racial representation, the
identification probability for White Americans is significantly higher
than that for Black Americans (see Figures 1 and 2 in Section 4.2),
primarily due to disparities in database representation.</p></li>
<li><p>The value of <span
class="math inline">\(P(\text{identify}|\bar{r})\)</span> is notably
higher than <span
class="math inline">\(E[P(\text{identify}|r)]\)</span>, highlighting the
importance of considering family size as a complete distribution in the
analysis.</p></li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Detroit
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.5.1   workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] gtable_0.3.6      jsonlite_1.8.9    compiler_4.4.2    promises_1.3.2   
 [5] Rcpp_1.0.13-1     stringr_1.5.1     git2r_0.35.0      callr_3.7.6      
 [9] later_1.4.1       jquerylib_0.1.4   scales_1.3.0      yaml_2.3.10      
[13] fastmap_1.2.0     R6_2.5.1          labeling_0.4.3    knitr_1.49       
[17] tibble_3.2.1      munsell_0.5.1     rprojroot_2.0.4   bslib_0.8.0      
[21] pillar_1.9.0      rlang_1.1.4       utf8_1.2.4        cachem_1.1.0     
[25] stringi_1.8.4     httpuv_1.6.15     xfun_0.49         getPass_0.2-4    
[29] fs_1.6.5          sass_0.4.9        cli_3.6.3         withr_3.0.2      
[33] magrittr_2.0.3    ps_1.8.1          grid_4.4.2        digest_0.6.37    
[37] processx_3.8.4    rstudioapi_0.17.1 lifecycle_1.0.4   vctrs_0.6.5      
[41] evaluate_1.0.1    glue_1.8.0        farver_2.1.2      whisker_0.4.1    
[45] colorspace_2.1-1  fansi_1.0.6       rmarkdown_2.29    httr_1.4.7       
[49] tools_4.4.2       pkgconfig_2.0.3   htmltools_0.5.8.1</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
