<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tina Lasisi" />


<title>Simulation of STR Pairs and Calculation of Likelihood Ratios</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PODFRIDGE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
<li>
  <a href="demo-data.html">Demographic Data</a>
</li>
<li>
  <a href="database-composition.html">Database composition</a>
</li>
<li>
  <a href="STR-simulation.html">STR simulations</a>
</li>
<li>
  <a href="CODIS-STR-freqs.html">CODIS STRs</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tinalasisi/PODFRIDGE">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Simulation of STR Pairs and Calculation of
Likelihood Ratios</h1>
<h4 class="author">Tina Lasisi</h4>
<h4 class="date">2024-07-19 18:23:50</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-19
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>PODFRIDGE/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230302code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230302code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230302)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlasisilabPODFRIDGEtree635de08209d7930a6436c8693341cf7793cdaf84targetblank635de08a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lasisilab/PODFRIDGE/tree/635de08209d7930a6436c8693341cf7793cdaf84" target="_blank">635de08</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlasisilabPODFRIDGEtree635de08209d7930a6436c8693341cf7793cdaf84targetblank635de08a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lasisilab/PODFRIDGE/tree/635de08209d7930a6436c8693341cf7793cdaf84" target="_blank">635de08</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  data/core_CODIS_loci.csv
    Untracked:  data/df_allelefreq.csv

Unstaged changes:
    Modified:   analysis/STR-simulation.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/STR-simulation.Rmd</code>) and
HTML (<code>docs/STR-simulation.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/635de08209d7930a6436c8693341cf7793cdaf84/analysis/STR-simulation.Rmd" target="_blank">635de08</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-18
</td>
<td>
Updating html for sibling analysis and STR simulations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/635de08209d7930a6436c8693341cf7793cdaf84/docs/STR-simulation.html" target="_blank">635de08</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-18
</td>
<td>
Updating html for sibling analysis and STR simulations
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/c57a79a3775c61c5ff042bb3adf31a0c8fb778f1/analysis/STR-simulation.Rmd" target="_blank">c57a79a</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-10
</td>
<td>
Updated STR-simulation.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/c57a79a3775c61c5ff042bb3adf31a0c8fb778f1/docs/STR-simulation.html" target="_blank">c57a79a</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-10
</td>
<td>
Updated STR-simulation.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/f80f86ef4c12e38ac5c8d73fd3561ebcc831329c/analysis/STR-simulation.Rmd" target="_blank">f80f86e</a>
</td>
<td>
tinalasisi
</td>
<td>
2024-07-10
</td>
<td>
Updated STR-simulation.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/f80f86ef4c12e38ac5c8d73fd3561ebcc831329c/docs/STR-simulation.html" target="_blank">f80f86e</a>
</td>
<td>
tinalasisi
</td>
<td>
2024-07-10
</td>
<td>
Updated STR-simulation.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/11fb32c8db06273cc164a483beec8755b89037d1/analysis/STR-simulation.Rmd" target="_blank">11fb32c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-10
</td>
<td>
write CSVs with output data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/cf281b698161b692f571b83e0d7a11f828eb6a91/docs/STR-simulation.html" target="_blank">cf281b6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/2596546a55e9369c81787cb7f9f35c660272aa16/analysis/STR-simulation.Rmd" target="_blank">2596546</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
<td>
wflow_publish("analysis/*", republish = TRUE, all = TRUE, verbose =
TRUE)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="from-weight-of-evidence-for-forensic-dna-profiles-book"
class="section level3">
<h3>From Weight-of-evidence for forensic DNA profiles book</h3>
<p>Likelihood ratio for a single locus is:</p>
<p><span class="math display">\[
R=\kappa_0+\kappa_1 / R_X^p+\kappa_2 / R_X^u
\]</span> Where <span class="math inline">\(\kappa\)</span> is the
probability of having 0, 1 or 2 alleles IBD for a given
relationship.</p>
<p>The <span class="math inline">\(R_X\)</span> terms are quantifying
the “surprisingness” of a particular pattern of allele sharing.</p>
<p>The <span class="math inline">\(R_X^p\)</span> terms attached to the
<span class="math inline">\(kappa_1\)</span> are defined in the
following table:</p>
<p><span class="math display">\[
\begin{aligned}
&amp;\text { Table 7.2 Single-locus LRs for paternity when }
\mathcal{C}_M \text { is unavailable. }\\
&amp;\begin{array}{llc}
\hline c &amp; Q &amp; R_X \times\left(1+2 F_{S T}\right) \\
\hline \mathrm{AA} &amp; \mathrm{AA} &amp; 3 F_{S T}+\left(1-F_{S
T}\right) p_A \\
\mathrm{AA} &amp; \mathrm{AB} &amp; 2\left(2 F_{S T}+\left(1-F_{S
T}\right) p_A\right) \\
\mathrm{AB} &amp; \mathrm{AA} &amp; 2\left(2 F_{S T}+\left(1-F_{S
T}\right) p_A\right) \\
\mathrm{AB} &amp; \mathrm{AC} &amp; 4\left(F_{S T}+\left(1-F_{S
T}\right) p_A\right) \\
\mathrm{AB} &amp; \mathrm{AB} &amp; 4\left(F_{S T}+\left(1-F_{S
T}\right) p_A\right)\left(F_{S T}+\left(1-F_{S T}\right) p_B\right)
/\left(2 F_{S T}+\left(1-F_{S T}\right)\left(p_A+p_B\right)\right) \\
\hline
\end{array}
\end{aligned}
\]</span></p>
<p>For our purposes we will take out the <span
class="math inline">\(F_{S T}\)</span> values. So the table will be as
follows:</p>
<p><span class="math display">\[
\begin{aligned}
&amp;\begin{array}{llc}
\hline c &amp; Q &amp; R_X \\
\hline \mathrm{AA} &amp; \mathrm{AA} &amp; p_A \\
\mathrm{AA} &amp; \mathrm{AB} &amp; 2 p_A \\
\mathrm{AB} &amp; \mathrm{AA} &amp; 2p_A \\
\mathrm{AB} &amp; \mathrm{AC} &amp; 4p_A \\
\mathrm{AB} &amp; \mathrm{AB} &amp; 4 p_A p_B/(p_A+p_B) \\
\hline
\end{array}
\end{aligned}
\]</span></p>
<p>If none of the alleles match, then the <span
class="math inline">\(\kappa_1 / R_X^p = 0\)</span>.</p>
<p>The <span class="math inline">\(R_X^u\)</span> terms attached to the
<span class="math inline">\(kappa_2\)</span> are defined as:</p>
<p>If both alleles match and are homozygous the equation is 6.4 (pg 85).
Single locus match probability: <span
class="math inline">\(\mathrm{CSP}=\mathcal{G}_Q=\mathrm{AA}\)</span>
<span class="math display">\[
\frac{\left(2 F_{S T}+\left(1-F_{S T}\right) p_A\right)\left(3 F_{S
T}+\left(1-F_{S T}\right) p_A\right)}{\left(1+F_{S T}\right)\left(1+2
F_{S T}\right)}
\]</span> Simplified to: <span class="math display">\[
p_A{ }^2
\]</span></p>
<p>If both alleles match and are heterozygous, the equation is 6.5 (pg
85) Single locus match probability: <span
class="math inline">\(\mathrm{CSP}=\mathcal{G}_Q=\mathrm{AB}\)</span>
<span class="math display">\[
2 \frac{\left(F_{S T}+\left(1-F_{S T}\right) p_A\right)\left(F_{S
T}+\left(1-F_{S T}\right) p_B\right)}{\left(1+F_{S T}\right)\left(1+2
F_{S T}\right)}
\]</span> Simplified to:</p>
<p><span class="math display">\[
2 p_A p_B
\]</span> If both alleles do not match then <span
class="math inline">\(\kappa_2 / R_X^u = 0\)</span>.</p>
</div>
<div id="lr-function" class="section level2">
<h2>LR function</h2>
<p>Flowchart</p>
<p><img src="assets/STR_LR.png" /> ## Likelihood ratio funtion</p>
<pre class="r"><code>calculate_likelihood_ratio &lt;- function(shared_alleles, genotype_match = NULL, pA = NULL, pB = NULL, k0, k1, k2) {

  # Case 0: No Shared Alleles
  if (shared_alleles == 0) {
    LR &lt;- k0
    return(LR)
  }

  # Case 1: One Shared Allele
  if (shared_alleles == 1) {
    if (genotype_match == &quot;AA-AA&quot;) {
      Rxp &lt;- pA
    } else if (genotype_match == &quot;AA-AB&quot; | genotype_match == &quot;AB-AA&quot;) {
      Rxp &lt;- 2 * pA
    } else if (genotype_match == &quot;AB-AC&quot;) {
      Rxp &lt;- 4 * pA
    } else if (genotype_match == &quot;AB-AB&quot;) {
      Rxp &lt;- (4 * pA * pB) / (pA + pB)
    } else {
      stop(&quot;Invalid genotype match for 1 shared allele.&quot;)
    }
    
    LR &lt;- k0 + (k1 / Rxp)
    return(LR)
  }

  # Case 2: Two Shared Alleles
  if (shared_alleles == 2) {
    if (genotype_match == &quot;AA-AA&quot;) {
      Rxp &lt;- pA
      Rxu &lt;- pA^2
    } else if (genotype_match == &quot;AB-AB&quot;) {
      Rxp &lt;- (4 * pA * pB) / (pA + pB)
      Rxu &lt;- 2 * pA * pB
    } else {
      stop(&quot;Invalid genotype match for 2 shared alleles.&quot;)
    }

    LR &lt;- k0 + (k1 / Rxp) + (k2 / Rxu)
    return(LR)
  }
}</code></pre>
<p>Input for LR function</p>
</div>
<div id="rscript-for-simulation" class="section level2">
<h2>Rscript for simulation</h2>
<p><img src="assets/simulation-framework.png" /> ### Step 1: Simulation
Setup</p>
<p>The first step is to set up the simulation by defining the parameters
and functions required for generating the simulated data. This includes
specifying the population size, allele frequencies, and the kinship
matrix for likelihood ratio calculations. The simulation will generate
pairs of individuals with known relationships (e.g., parent-child,
siblings) and unknown relationships (unrelated) based on the specified
parameters.</p>
<pre class="r"><code>generate_simulation_setup &lt;- function(kinship_matrix, population_list, num_related, num_unrelated) {
  # Create an empty dataframe to store the simulation setup
  simulation_setup &lt;- data.frame(
    population = character(),
    relationship_type = character(),
    num_simulations = integer(),
    stringsAsFactors = FALSE
  )
  
  # Loop through each population and relationship type to create the setup
  for (population in population_list) {
    for (relationship in kinship_matrix$relationship_type) {
      num_simulations &lt;- ifelse(relationship == &quot;unrelated&quot;, num_unrelated, num_related)
      
      # Append to the simulation setup dataframe
      simulation_setup &lt;- rbind(simulation_setup, data.frame(
        population = population,
        relationship_type = relationship,
        num_simulations = num_simulations
      ))
    }
  }
  
  return(simulation_setup)
}</code></pre>
<p>The necessary parameters for this are initialized beforehand.</p>
<p>First, the kinship coefficients are provided in a matrix:</p>
<pre class="r"><code># Create a dataframe with relationship types and their respective kinship coefficients (k0, k1, k2)
kinship_matrix &lt;- tibble(
    relationship_type =
      c(&quot;parent_child&quot;, &quot;full_siblings&quot;, &quot;half_siblings&quot;, &quot;cousins&quot;, &quot;second_cousins&quot;, &quot;unrelated&quot;),
    k0 = c(0, 1/4, 1/2, 7/8, 15/16, 1),
    k1 = c(1, 1/2, 1/2, 1/8, 1/16, 0),
    k2 = c(0, 1/4, 0, 0, 0, 0)
  )

# Print the kinship matrix to check the contents
print(kinship_matrix)</code></pre>
<pre><code># A tibble: 6 × 4
  relationship_type    k0     k1    k2
  &lt;chr&gt;             &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;
1 parent_child      0     1       0   
2 full_siblings     0.25  0.5     0.25
3 half_siblings     0.5   0.5     0   
4 cousins           0.875 0.125   0   
5 second_cousins    0.938 0.0625  0   
6 unrelated         1     0       0   </code></pre>
<p>Then a list of populations is created:</p>
<pre><code>[1] &quot;all&quot;      &quot;AfAm&quot;     &quot;Cauc&quot;     &quot;Hispanic&quot; &quot;Asian&quot;   </code></pre>
<p>Now we can test the simulation setup function:</p>
<pre><code>   population relationship_type num_simulations
1         all      parent_child               2
2         all     full_siblings               2
3         all     half_siblings               2
4         all           cousins               2
5         all    second_cousins               2
6         all         unrelated               5
7        AfAm      parent_child               2
8        AfAm     full_siblings               2
9        AfAm     half_siblings               2
10       AfAm           cousins               2
11       AfAm    second_cousins               2
12       AfAm         unrelated               5
13       Cauc      parent_child               2
14       Cauc     full_siblings               2
15       Cauc     half_siblings               2
16       Cauc           cousins               2
17       Cauc    second_cousins               2
18       Cauc         unrelated               5
19   Hispanic      parent_child               2
20   Hispanic     full_siblings               2
21   Hispanic     half_siblings               2
22   Hispanic           cousins               2
23   Hispanic    second_cousins               2
24   Hispanic         unrelated               5
25      Asian      parent_child               2
26      Asian     full_siblings               2
27      Asian     half_siblings               2
28      Asian           cousins               2
29      Asian    second_cousins               2
30      Asian         unrelated               5</code></pre>
<div id="step-2-initialize-individuals" class="section level3">
<h3>Step 2: Initialize Individuals</h3>
<p>Once the general simulation setup is done, each pair of individuals
must be initialized as its own dataframe. This requires knowing how many
loci will be simulated per pair. We will therefore first extract the
unique loci from our allele frequency tables.</p>
<pre><code>Rows: 387 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (2): marker, population
dbl (2): allele, frequency

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 453 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (2): marker, population
dbl (2): allele, frequency

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 277 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (2): marker, population
dbl (2): allele, frequency

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 344 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (2): marker, population
dbl (2): allele, frequency

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 348 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (2): marker, population
dbl (2): allele, frequency

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre><code># A tibble: 1,809 × 4
   marker  allele frequency population
   &lt;chr&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;     
 1 Penta_D    2.2   0.114   AfAm      
 2 F13A01     3.2   0.118   AfAm      
 3 Penta_D    3.2   0.00877 AfAm      
 4 F13A01     4     0.0687  AfAm      
 5 F13A01     4.2   0.00146 AfAm      
 6 D16S539    5     0.00146 AfAm      
 7 F13A01     5     0.339   AfAm      
 8 Penta_C    5     0.0249  AfAm      
 9 Penta_D    5     0.0439  AfAm      
10 Penta_E    5     0.0950  AfAm      
# ℹ 1,799 more rows</code></pre>
<p>We will now extract the unique loci from the allele frequency
tables:</p>
<pre><code> [1] &quot;Penta_D&quot;  &quot;F13A01&quot;   &quot;D16S539&quot;  &quot;Penta_C&quot;  &quot;Penta_E&quot;  &quot;TH01&quot;    
 [7] &quot;D7S820&quot;   &quot;F13B&quot;     &quot;FESFPS&quot;   &quot;TPOX&quot;     &quot;CSF1PO&quot;   &quot;D5S818&quot;  
[13] &quot;LPL&quot;      &quot;SE33&quot;     &quot;D10S1248&quot; &quot;D13S317&quot;  &quot;D22S1045&quot; &quot;D8S1179&quot; 
[19] &quot;D18S51&quot;   &quot;D2S441&quot;   &quot;D6S1043&quot;  &quot;D19S433&quot;  &quot;D1S1656&quot;  &quot;vWA&quot;     
[25] &quot;D3S1358&quot;  &quot;D12S391&quot;  &quot;D2S1338&quot;  &quot;FGA&quot;      &quot;D21S11&quot;  </code></pre>
<pre><code>Number of unique loci: 29 </code></pre>
<p>These are the 29 autosomal loci from the 2013 and 2017 FSI paper on
US STR allele frequencies for 29 autosomal STR loci <a
href="https://www.fsigenetics.com/article/S1872-4973(17)30180-1/fulltext">Steffen
et al 2017</a>.</p>
<p>We will use the list of different required loci to calculate
likelihood ratios for pairs of individuals. Below is a reference with
which loci are used in various sets.</p>
<pre><code>      locus core_13 identifiler_15 expanded_20 supplementary
1    CSF1PO       1              1           1             1
2       FGA       1              1           1             1
3      THO1       1              1           1             1
4      TPOX       1              1           1             1
5       vWA       1              1           1             1
6   D3S1358       1              1           1             1
7    D5S818       1              1           1             1
8    D7S820       1              1           1             1
9   D8S1179       1              1           1             1
10  D13S317       1              1           1             1
11  D16S539       1              1           1             1
12   D18S51       1              1           1             1
13   D21S11       1              1           1             1
14  D1S1656       0              0           1             1
15   D2S441       0              0           1             1
16  D2S1338       0              1           1             1
17 D10S1248       0              0           1             1
18  D12S391       0              0           1             1
19  D19S433       0              1           1             1
20 D22S1045       0              0           1             1
21     SE33       0              0           0             1
22  Penta_E       0              0           0             1
23  Penta_D       0              0           0             1</code></pre>
<pre><code>$core_13
 [1] &quot;CSF1PO&quot;  &quot;FGA&quot;     &quot;THO1&quot;    &quot;TPOX&quot;    &quot;vWA&quot;     &quot;D3S1358&quot; &quot;D5S818&quot; 
 [8] &quot;D7S820&quot;  &quot;D8S1179&quot; &quot;D13S317&quot; &quot;D16S539&quot; &quot;D18S51&quot;  &quot;D21S11&quot; 

$identifiler_15
 [1] &quot;CSF1PO&quot;  &quot;FGA&quot;     &quot;THO1&quot;    &quot;TPOX&quot;    &quot;vWA&quot;     &quot;D3S1358&quot; &quot;D5S818&quot; 
 [8] &quot;D7S820&quot;  &quot;D8S1179&quot; &quot;D13S317&quot; &quot;D16S539&quot; &quot;D18S51&quot;  &quot;D21S11&quot;  &quot;D2S1338&quot;
[15] &quot;D19S433&quot;

$expanded_20
 [1] &quot;CSF1PO&quot;   &quot;FGA&quot;      &quot;THO1&quot;     &quot;TPOX&quot;     &quot;vWA&quot;      &quot;D3S1358&quot; 
 [7] &quot;D5S818&quot;   &quot;D7S820&quot;   &quot;D8S1179&quot;  &quot;D13S317&quot;  &quot;D16S539&quot;  &quot;D18S51&quot;  
[13] &quot;D21S11&quot;   &quot;D1S1656&quot;  &quot;D2S441&quot;   &quot;D2S1338&quot;  &quot;D10S1248&quot; &quot;D12S391&quot; 
[19] &quot;D19S433&quot;  &quot;D22S1045&quot;

$supplementary
 [1] &quot;CSF1PO&quot;   &quot;FGA&quot;      &quot;THO1&quot;     &quot;TPOX&quot;     &quot;vWA&quot;      &quot;D3S1358&quot; 
 [7] &quot;D5S818&quot;   &quot;D7S820&quot;   &quot;D8S1179&quot;  &quot;D13S317&quot;  &quot;D16S539&quot;  &quot;D18S51&quot;  
[13] &quot;D21S11&quot;   &quot;D1S1656&quot;  &quot;D2S441&quot;   &quot;D2S1338&quot;  &quot;D10S1248&quot; &quot;D12S391&quot; 
[19] &quot;D19S433&quot;  &quot;D22S1045&quot; &quot;SE33&quot;     &quot;Penta_E&quot;  &quot;Penta_D&quot; 

$autosomal_29
 [1] &quot;Penta_D&quot;  &quot;F13A01&quot;   &quot;D16S539&quot;  &quot;Penta_C&quot;  &quot;Penta_E&quot;  &quot;TH01&quot;    
 [7] &quot;D7S820&quot;   &quot;F13B&quot;     &quot;FESFPS&quot;   &quot;TPOX&quot;     &quot;CSF1PO&quot;   &quot;D5S818&quot;  
[13] &quot;LPL&quot;      &quot;SE33&quot;     &quot;D10S1248&quot; &quot;D13S317&quot;  &quot;D22S1045&quot; &quot;D8S1179&quot; 
[19] &quot;D18S51&quot;   &quot;D2S441&quot;   &quot;D6S1043&quot;  &quot;D19S433&quot;  &quot;D1S1656&quot;  &quot;vWA&quot;     
[25] &quot;D3S1358&quot;  &quot;D12S391&quot;  &quot;D2S1338&quot;  &quot;FGA&quot;      &quot;D21S11&quot;  </code></pre>
<p>Now we can initialize the individuals:</p>
<pre><code>[1] &quot;Initialized individuals:&quot;</code></pre>
<pre><code>   population relationship_type sim_id    locus ind1_allele1 ind1_allele2
1         all      parent_child      1  Penta_D                          
2         all      parent_child      1   F13A01                          
3         all      parent_child      1  D16S539                          
4         all      parent_child      1  Penta_C                          
5         all      parent_child      1  Penta_E                          
6         all      parent_child      1     TH01                          
7         all      parent_child      1   D7S820                          
8         all      parent_child      1     F13B                          
9         all      parent_child      1   FESFPS                          
10        all      parent_child      1     TPOX                          
11        all      parent_child      1   CSF1PO                          
12        all      parent_child      1   D5S818                          
13        all      parent_child      1      LPL                          
14        all      parent_child      1     SE33                          
15        all      parent_child      1 D10S1248                          
16        all      parent_child      1  D13S317                          
17        all      parent_child      1 D22S1045                          
18        all      parent_child      1  D8S1179                          
19        all      parent_child      1   D18S51                          
20        all      parent_child      1   D2S441                          
21        all      parent_child      1  D6S1043                          
22        all      parent_child      1  D19S433                          
23        all      parent_child      1  D1S1656                          
24        all      parent_child      1      vWA                          
25        all      parent_child      1  D3S1358                          
26        all      parent_child      1  D12S391                          
27        all      parent_child      1  D2S1338                          
28        all      parent_child      1      FGA                          
29        all      parent_child      1   D21S11                          
   ind2_allele1 ind2_allele2 shared_alleles genotype_match LR
1                                         0                 0
2                                         0                 0
3                                         0                 0
4                                         0                 0
5                                         0                 0
6                                         0                 0
7                                         0                 0
8                                         0                 0
9                                         0                 0
10                                        0                 0
11                                        0                 0
12                                        0                 0
13                                        0                 0
14                                        0                 0
15                                        0                 0
16                                        0                 0
17                                        0                 0
18                                        0                 0
19                                        0                 0
20                                        0                 0
21                                        0                 0
22                                        0                 0
23                                        0                 0
24                                        0                 0
25                                        0                 0
26                                        0                 0
27                                        0                 0
28                                        0                 0
29                                        0                 0</code></pre>
<pre><code>  population relationship_type sim_id  locus ind1_allele1 ind1_allele2
9        all      parent_child      1 FESFPS           11         10.3
  ind2_allele1 ind2_allele2 shared_alleles genotype_match LR
9         10.3           13              0                 0</code></pre>
<pre><code>  population relationship_type sim_id  locus ind1_allele1 ind1_allele2
9        all      parent_child      1 FESFPS           11         10.3
  ind2_allele1 ind2_allele2 shared_alleles genotype_match       LR
9         10.3           13              1          AB-AC 16.70968</code></pre>
<pre><code>   population relationship_type sim_id    locus ind1_allele1 ind1_allele2
1         all      parent_child      1  Penta_D           13           12
2         all      parent_child      1   F13A01            7            5
3         all      parent_child      1  D16S539            9           12
4         all      parent_child      1  Penta_C           14           11
5         all      parent_child      1  Penta_E           11           13
6         all      parent_child      1     TH01            6            7
7         all      parent_child      1   D7S820           11            9
8         all      parent_child      1     F13B           10           10
9         all      parent_child      1   FESFPS           12           11
10        all      parent_child      1     TPOX            6            8
11        all      parent_child      1   CSF1PO           10           10
12        all      parent_child      1   D5S818           10           13
13        all      parent_child      1      LPL           10           11
14        all      parent_child      1     SE33         25.2           17
15        all      parent_child      1 D10S1248           13           14
16        all      parent_child      1  D13S317           11           12
17        all      parent_child      1 D22S1045           15           17
18        all      parent_child      1  D8S1179           16           12
19        all      parent_child      1   D18S51           15           13
20        all      parent_child      1   D2S441           12           14
21        all      parent_child      1  D6S1043           13           19
22        all      parent_child      1  D19S433           15           14
23        all      parent_child      1  D1S1656         16.3           16
24        all      parent_child      1      vWA           14           17
25        all      parent_child      1  D3S1358           17           18
26        all      parent_child      1  D12S391           16           17
27        all      parent_child      1  D2S1338           19           22
28        all      parent_child      1      FGA           22           18
29        all      parent_child      1   D21S11         30.2           29
   ind2_allele1 ind2_allele2 shared_alleles genotype_match        LR
1            13            8              1          AB-AC 1.8031359
2             5            6              1          AB-AC 1.0882353
3            11           12              1          AB-AC 0.9736842
4            13           11              1          AB-AC 0.7485549
5            14           13              1          AB-AC 2.7700535
6             9            6              1          AB-AC 1.2758621
7            11            9              2          AB-AB 3.1213992
8             9           10              1          AA-AB 1.4428969
9            12           11              2          AB-AB 1.7009816
10           11            6              1          AB-AC 7.8409091
11           12           10              1          AA-AB 2.1538462
12           13           12              1          AB-AC 1.5325444
13           11           11              1          AB-AA 2.6361323
14           17         29.2              1          AB-AC 3.4078947
15           13           13              1          AB-AA 1.8080279
16           11           11              1          AB-AA 1.7209302
17           15           15              1          AB-AA 1.5578947
18           15           16              1          AB-AC 5.1287129
19           13           12              1          AB-AC 2.3870968
20           14           12              2          AB-AB 3.6166908
21           12           13              1          AB-AC 2.5643564
22           15           15              1          AB-AA 4.2459016
23         17.3         16.3              1          AB-AC 3.6737589
24           17           15              1          AB-AC 0.9539595
25           18           16              1          AB-AC 2.3652968
26           16           18              1          AB-AC 6.1666667
27           20           19              1          AB-AC 1.6818182
28           23           22              1          AB-AC 1.2665037
29           31           29              1          AB-AC 1.2245863</code></pre>
<pre><code>Execution time for list_of_rows_approach:</code></pre>
<pre><code>   user  system elapsed 
  0.529   0.010   1.393 </code></pre>
<pre><code>Execution time for future_pmap_approach:</code></pre>
<pre><code>   user  system elapsed 
  0.528   0.009   0.571 </code></pre>
<pre><code>Benchmark results for both approaches:</code></pre>
<pre><code>Unit: milliseconds
         expr      min       lq     mean   median      uq      max neval
 list_of_rows 529.5072 531.5589 545.7189 541.9877 552.289 579.0596    10
  future_pmap 526.2015 531.2727 539.3893 536.4072 545.551 559.4490    10</code></pre>
</div>
</div>
<div id="simulation-results" class="section level2">
<h2>Simulation results</h2>
<pre><code>Rows: 360000 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (3): population, known_relationship_type, tested_relationship_type
dbl (3): replicate_id, num_shared_alleles_sum, log_R_sum

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div
id="proportion-of-individuals-of-known-relationship-type-exceeding-likelihood-cut-off"
class="section level2">
<h2>Proportion of individuals of known relationship type exceeding
likelihood cut-off</h2>
<pre><code>  population relationship_type fp_rate prop_exceeding
1       AfAm      parent_child     0.1          1.000
2      Asian      parent_child     0.1          1.000
3       Cauc      parent_child     0.1          1.000
4   Hispanic      parent_child     0.1          1.000
5       AfAm     full_siblings     0.1          0.794
6      Asian     full_siblings     0.1          0.820</code></pre>
<pre><code>`summarise()` has grouped output by &#39;population&#39;. You can override using the
`.groups` argument.</code></pre>
<p><img src="figure/STR-simulation.Rmd/unnamed-chunk-23-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cut-offs-for-each-fpr" class="section level2">
<h2>Cut-offs for each FPR</h2>
<pre><code>`summarise()` has grouped output by &#39;population&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code># A tibble: 24 × 7
   population known_relationship_type mean_log_R_sum median_log_R_sum
   &lt;chr&gt;      &lt;chr&gt;                            &lt;dbl&gt;            &lt;dbl&gt;
 1 AfAm       cousins                         -0.533            2.42 
 2 AfAm       full_siblings                   10.5             11.5  
 3 AfAm       half_siblings                    3.85             6.80 
 4 AfAm       parent_child                    19.6             19.0  
 5 AfAm       second_cousins                  -1.23             1.78 
 6 AfAm       unrelated                       -2.00             0.914
 7 Asian      cousins                         -0.538            2.42 
 8 Asian      full_siblings                   10.5             11.6  
 9 Asian      half_siblings                    3.82             6.78 
10 Asian      parent_child                    19.5             18.9  
# ℹ 14 more rows
# ℹ 3 more variables: min_log_R_sum &lt;dbl&gt;, max_log_R_sum &lt;dbl&gt;, count &lt;int&gt;</code></pre>
<div
id="calculate-llr-cutoffs-for-unrelated-pairs-at-each-fpr-by-population"
class="section level4">
<h4>Calculate LLR cutoffs for unrelated pairs at each FPR by
population</h4>
<p>This table presents aggregated log_R_sum values by relationship type
for each population, alongside LLR cutoffs calculated for unrelated
pairs to maintain false positive rates (FPRs) of 1%, 0.1%, and 0.01%.
The cutoffs indicate the LLR threshold above which a pair is less likely
to be unrelated at the given FPR, thereby serving as a critical
benchmark for assessing relationship evidence. The mean log_R_sum values
further elucidate the average strength of genetic evidence supporting
each relationship type within populations, highlighting variations and
consistencies in genetic relatedness indicators across demographic
groups.</p>
<pre><code>
Attaching package: &#39;kableExtra&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    group_rows</code></pre>
<table class="table table-striped table-hover table-condensed" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
population
</th>
<th style="text-align:right;">
cousins
</th>
<th style="text-align:right;">
full_siblings
</th>
<th style="text-align:right;">
half_siblings
</th>
<th style="text-align:right;">
parent_child
</th>
<th style="text-align:right;">
second_cousins
</th>
<th style="text-align:right;">
unrelated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
AfAm
</td>
<td style="text-align:right;">
-0.5334790
</td>
<td style="text-align:right;">
10.45417
</td>
<td style="text-align:right;">
3.849582
</td>
<td style="text-align:right;">
19.62211
</td>
<td style="text-align:right;">
-1.229432
</td>
<td style="text-align:right;">
-2.003701
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Asian
</td>
<td style="text-align:right;">
-0.5379709
</td>
<td style="text-align:right;">
10.45853
</td>
<td style="text-align:right;">
3.821000
</td>
<td style="text-align:right;">
19.54267
</td>
<td style="text-align:right;">
-1.387000
</td>
<td style="text-align:right;">
-1.982983
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Cauc
</td>
<td style="text-align:right;">
-0.5075911
</td>
<td style="text-align:right;">
10.48275
</td>
<td style="text-align:right;">
3.873924
</td>
<td style="text-align:right;">
19.51039
</td>
<td style="text-align:right;">
-1.218428
</td>
<td style="text-align:right;">
-1.982207
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Hispanic
</td>
<td style="text-align:right;">
-0.5536143
</td>
<td style="text-align:right;">
10.40972
</td>
<td style="text-align:right;">
3.837237
</td>
<td style="text-align:right;">
19.58039
</td>
<td style="text-align:right;">
-1.249979
</td>
<td style="text-align:right;">
-1.984905
</td>
</tr>
</tbody>
</table>
<p><img src="figure/STR-simulation.Rmd/unnamed-chunk-31-1.png" width="768" style="display: block; margin: auto;" /></p>
<p><img src="figure/STR-simulation.Rmd/unnamed-chunk-33-1.png" width="768" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-apple-darwin20
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Detroit
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] kableExtra_1.4.0      furrr_0.3.1           future_1.33.2        
 [4] microbenchmark_1.4.10 patchwork_1.2.0       lubridate_1.9.3      
 [7] forcats_1.0.0         stringr_1.5.1         dplyr_1.1.4          
[10] purrr_1.0.2           readr_2.1.5           tidyr_1.3.1          
[13] tibble_3.2.1          ggplot2_3.5.1         tidyverse_2.0.0      
[16] RColorBrewer_1.1-3    wesanderson_0.3.7     workflowr_1.7.1      

loaded via a namespace (and not attached):
 [1] gtable_0.3.5      xfun_0.45         bslib_0.7.0       processx_3.8.4   
 [5] callr_3.7.6       tzdb_0.4.0        vctrs_0.6.5       tools_4.4.1      
 [9] ps_1.7.6          generics_0.1.3    parallel_4.4.1    fansi_1.0.6      
[13] highr_0.11        pkgconfig_2.0.3   lifecycle_1.0.4   farver_2.1.2     
[17] compiler_4.4.1    git2r_0.33.0      munsell_0.5.1     getPass_0.2-4    
[21] codetools_0.2-20  httpuv_1.6.15     htmltools_0.5.8.1 sass_0.4.9       
[25] yaml_2.3.8        crayon_1.5.3      later_1.3.2       pillar_1.9.0     
[29] jquerylib_0.1.4   whisker_0.4.1     cachem_1.1.0      parallelly_1.37.1
[33] tidyselect_1.2.1  digest_0.6.36     stringi_1.8.4     listenv_0.9.1    
[37] labeling_0.4.3    rprojroot_2.0.4   fastmap_1.2.0     grid_4.4.1       
[41] colorspace_2.1-0  cli_3.6.3         magrittr_2.0.3    utf8_1.2.4       
[45] withr_3.0.0       scales_1.3.0      promises_1.3.0    bit64_4.0.5      
[49] timechange_0.3.0  rmarkdown_2.27    httr_1.4.7        globals_0.16.3   
[53] bit_4.0.5         hms_1.1.3         evaluate_0.24.0   knitr_1.47       
[57] viridisLite_0.4.2 rlang_1.1.4       Rcpp_1.0.12       glue_1.7.0       
[61] xml2_1.3.6        svglite_2.1.3     vroom_1.6.5       rstudioapi_0.16.0
[65] jsonlite_1.8.8    R6_2.5.1          systemfonts_1.1.0 fs_1.6.4         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
