<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tina Lasisi" />


<title>Simulation of STR Pairs and Calculation of Likelihood Ratios</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PODFRIDGE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lasisilab/PODFRIDGE">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Simulation of STR Pairs and Calculation of
Likelihood Ratios</h1>
<h4 class="author">Tina Lasisi</h4>
<h4 class="date">2025-02-08 19:24:32</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>PODFRIDGE/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230302code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230302code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230302)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlasisilabPODFRIDGEtree669998c3574753cb3e9d68c015179304b80a9bbbtargetblank669998ca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lasisilab/PODFRIDGE/tree/669998c3574753cb3e9d68c015179304b80a9bbb" target="_blank">669998c</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlasisilabPODFRIDGEtree669998c3574753cb3e9d68c015179304b80a9bbbtargetblank669998ca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lasisilab/PODFRIDGE/tree/669998c3574753cb3e9d68c015179304b80a9bbb" target="_blank">669998c</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/sims/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/simulation_20240726-155743/.DS_Store
    Ignored:    output/simulation_20240726-162034_11228488/.DS_Store
    Ignored:    output/simulation_20240726-163235_11228791/.DS_Store

Unstaged changes:
    Modified:   PODFRIDGE.Rproj
    Modified:   analysis/STR-simulation.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/STR-simulation.Rmd</code>) and
HTML (<code>docs/STR-simulation.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/f143ee1386425f8e06d3a94f09d512ac794e407f/docs/STR-simulation.html" target="_blank">f143ee1</a>
</td>
<td>
tinalasisi
</td>
<td>
2024-09-16
</td>
<td>
Revised website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/b98be9778d2e820b4bc587f3596d68fa71e5a53b/docs/STR-simulation.html" target="_blank">b98be97</a>
</td>
<td>
linmatch
</td>
<td>
2024-08-30
</td>
<td>
update color figure 1,2
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/36cbb65c6e92da91aaa68d90fcd797f82a43525a/analysis/STR-simulation.Rmd" target="_blank">36cbb65</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-25
</td>
<td>
Updated simulation scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/e20d18e7ff5470084d569930084bab11ab8c55ac/analysis/STR-simulation.Rmd" target="_blank">e20d18e</a>
</td>
<td>
tinalasisi
</td>
<td>
2024-07-25
</td>
<td>
Adding numbers and fixing figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/d15870a4cf8a28e306c281e19d41936eb2d6fcc0/analysis/STR-simulation.Rmd" target="_blank">d15870a</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-24
</td>
<td>
Setting up cluster instructions.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/26727407de579afe68bb37219862cfa5bb853129/analysis/STR-simulation.Rmd" target="_blank">2672740</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-24
</td>
<td>
Benchmarked dplyr and datatable versions of simulation
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/59a752d20b6109c34708818b839f48ec99366b57/analysis/STR-simulation.Rmd" target="_blank">59a752d</a>
</td>
<td>
tinalasisi
</td>
<td>
2024-07-24
</td>
<td>
Fixing simulation
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/68137c0e9e33a757f9cbf0bc3c8d890318bb7a7b/analysis/STR-simulation.Rmd" target="_blank">68137c0</a>
</td>
<td>
tinalasisi
</td>
<td>
2024-07-23
</td>
<td>
Benchmarking simulation script components
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/fbdfcf1e0d6544306069114e6a1850d8311ef93d/analysis/STR-simulation.Rmd" target="_blank">fbdfcf1</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-23
</td>
<td>
Fixed allele frequency table and continued simulation development
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/fbdfcf1e0d6544306069114e6a1850d8311ef93d/docs/STR-simulation.html" target="_blank">fbdfcf1</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-23
</td>
<td>
Fixed allele frequency table and continued simulation development
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/e1eec3c686364a517c3bc660f26678019d11a522/analysis/STR-simulation.Rmd" target="_blank">e1eec3c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-22
</td>
<td>
Updating the STR-simulation
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/b5e4ed4b749b7a2ae3f8a8ab144de66db2850e90/analysis/STR-simulation.Rmd" target="_blank">b5e4ed4</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-19
</td>
<td>
Updated simulation components + added to data folder
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/b5e4ed4b749b7a2ae3f8a8ab144de66db2850e90/docs/STR-simulation.html" target="_blank">b5e4ed4</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-19
</td>
<td>
Updated simulation components + added to data folder
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/635de08209d7930a6436c8693341cf7793cdaf84/analysis/STR-simulation.Rmd" target="_blank">635de08</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-18
</td>
<td>
Updating html for sibling analysis and STR simulations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/635de08209d7930a6436c8693341cf7793cdaf84/docs/STR-simulation.html" target="_blank">635de08</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-18
</td>
<td>
Updating html for sibling analysis and STR simulations
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/c57a79a3775c61c5ff042bb3adf31a0c8fb778f1/analysis/STR-simulation.Rmd" target="_blank">c57a79a</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-10
</td>
<td>
Updated STR-simulation.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/c57a79a3775c61c5ff042bb3adf31a0c8fb778f1/docs/STR-simulation.html" target="_blank">c57a79a</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-10
</td>
<td>
Updated STR-simulation.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/f80f86ef4c12e38ac5c8d73fd3561ebcc831329c/analysis/STR-simulation.Rmd" target="_blank">f80f86e</a>
</td>
<td>
tinalasisi
</td>
<td>
2024-07-10
</td>
<td>
Updated STR-simulation.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/f80f86ef4c12e38ac5c8d73fd3561ebcc831329c/docs/STR-simulation.html" target="_blank">f80f86e</a>
</td>
<td>
tinalasisi
</td>
<td>
2024-07-10
</td>
<td>
Updated STR-simulation.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/11fb32c8db06273cc164a483beec8755b89037d1/analysis/STR-simulation.Rmd" target="_blank">11fb32c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-10
</td>
<td>
write CSVs with output data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/cf281b698161b692f571b83e0d7a11f828eb6a91/docs/STR-simulation.html" target="_blank">cf281b6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/2596546a55e9369c81787cb7f9f35c660272aa16/analysis/STR-simulation.Rmd" target="_blank">2596546</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
<td>
wflow_publish("analysis/*", republish = TRUE, all = TRUE, verbose =
TRUE)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="background" class="section level2">
<h2>Background</h2>
<p>We will simulate pairs of individuals with known relationships (e.g.,
parent-child, siblings), including unrelated individuals, based on the
specified parameters. We will then calculate the likelihood ratio for
each pair of individuals based on the simulated genotypes and known
relationships.</p>
<p>To do this, we will draw the likelihood ratio caluclations from
Balding &amp; Steele’s ‘Weight-of-evidence for forensic DNA profiles’
book.</p>
<div id="from-weight-of-evidence-for-forensic-dna-profiles-book"
class="section level3">
<h3>From Weight-of-evidence for forensic DNA profiles book</h3>
<p>Likelihood ratio for a single locus is:</p>
<p><span class="math display">\[
R=\kappa_0+\kappa_1 / R_X^p+\kappa_2 / R_X^u
\]</span> Where <span class="math inline">\(\kappa\)</span> is the
probability of having 0, 1 or 2 alleles IBD for a given
relationship.</p>
<p>The <span class="math inline">\(R_X\)</span> terms are quantifying
the “surprisingness” of a particular pattern of allele sharing.</p>
<p>The <span class="math inline">\(R_X^p\)</span> terms attached to the
<span class="math inline">\(kappa_1\)</span> are defined in the
following table:</p>
<p><span class="math display">\[
\begin{aligned}
&amp;\text { Table 7.2 Single-locus LRs for paternity when }
\mathcal{C}_M \text { is unavailable. }\\
&amp;\begin{array}{llc}
\hline c &amp; Q &amp; R_X \times\left(1+2 F_{S T}\right) \\
\hline \mathrm{AA} &amp; \mathrm{AA} &amp; 3 F_{S T}+\left(1-F_{S
T}\right) p_A \\
\mathrm{AA} &amp; \mathrm{AB} &amp; 2\left(2 F_{S T}+\left(1-F_{S
T}\right) p_A\right) \\
\mathrm{AB} &amp; \mathrm{AA} &amp; 2\left(2 F_{S T}+\left(1-F_{S
T}\right) p_A\right) \\
\mathrm{AB} &amp; \mathrm{AC} &amp; 4\left(F_{S T}+\left(1-F_{S
T}\right) p_A\right) \\
\mathrm{AB} &amp; \mathrm{AB} &amp; 4\left(F_{S T}+\left(1-F_{S
T}\right) p_A\right)\left(F_{S T}+\left(1-F_{S T}\right) p_B\right)
/\left(2 F_{S T}+\left(1-F_{S T}\right)\left(p_A+p_B\right)\right) \\
\hline
\end{array}
\end{aligned}
\]</span></p>
<p>For our purposes we will take out the <span
class="math inline">\(F_{S T}\)</span> values. So the table will be as
follows:</p>
<p><span class="math display">\[
\begin{aligned}
&amp;\begin{array}{llc}
\hline c &amp; Q &amp; R_X \\
\hline \mathrm{AA} &amp; \mathrm{AA} &amp; p_A \\
\mathrm{AA} &amp; \mathrm{AB} &amp; 2 p_A \\
\mathrm{AB} &amp; \mathrm{AA} &amp; 2p_A \\
\mathrm{AB} &amp; \mathrm{AC} &amp; 4p_A \\
\mathrm{AB} &amp; \mathrm{AB} &amp; 4 p_A p_B/(p_A+p_B) \\
\hline
\end{array}
\end{aligned}
\]</span></p>
<p>If none of the alleles match, then the <span
class="math inline">\(\kappa_1 / R_X^p = 0\)</span>.</p>
<p>The <span class="math inline">\(R_X^u\)</span> terms attached to the
<span class="math inline">\(kappa_2\)</span> are defined as:</p>
<p>If both alleles match and are homozygous the equation is 6.4 (pg 85).
Single locus match probability: <span
class="math inline">\(\mathrm{CSP}=\mathcal{G}_Q=\mathrm{AA}\)</span>
<span class="math display">\[
\frac{\left(2 F_{S T}+\left(1-F_{S T}\right) p_A\right)\left(3 F_{S
T}+\left(1-F_{S T}\right) p_A\right)}{\left(1+F_{S T}\right)\left(1+2
F_{S T}\right)}
\]</span> Simplified to: <span class="math display">\[
p_A{ }^2
\]</span></p>
<p>If both alleles match and are heterozygous, the equation is 6.5 (pg
85) Single locus match probability: <span
class="math inline">\(\mathrm{CSP}=\mathcal{G}_Q=\mathrm{AB}\)</span>
<span class="math display">\[
2 \frac{\left(F_{S T}+\left(1-F_{S T}\right) p_A\right)\left(F_{S
T}+\left(1-F_{S T}\right) p_B\right)}{\left(1+F_{S T}\right)\left(1+2
F_{S T}\right)}
\]</span> Simplified to:</p>
<p><span class="math display">\[
2 p_A p_B
\]</span> If both alleles do not match then <span
class="math inline">\(\kappa_2 / R_X^u = 0\)</span>.</p>
</div>
</div>
<div id="likelihood-ratio-calculation" class="section level2">
<h2>Likelihood ratio calculation</h2>
<div id="flowchart" class="section level3">
<h3>Flowchart</h3>
<p>For our purposes, we can describe the flowchart for the likelihood
ratio calculation as follows:</p>
<p><img src="assets/LR-workflow.png" /></p>
</div>
<div id="function" class="section level3">
<h3>Function</h3>
<pre class="r"><code>calculate_likelihood_ratio &lt;- function(shared_alleles, genotype_match = NULL, pA = NULL, pB = NULL, k0, k1, k2) {

  # Case 0: No Shared Alleles
  if (shared_alleles == 0) {
    LR &lt;- k0
    return(LR)
  }

  # Case 1: One Shared Allele
  if (shared_alleles == 1) {
    if (genotype_match == &quot;AA-AA&quot;) {
      Rxp &lt;- pA
    } else if (genotype_match == &quot;AA-AB&quot; | genotype_match == &quot;AB-AA&quot;) {
      Rxp &lt;- 2 * pA
    } else if (genotype_match == &quot;AB-AC&quot;) {
      Rxp &lt;- 4 * pA
    } else if (genotype_match == &quot;AB-AB&quot;) {
      Rxp &lt;- (4 * (pA * pB) )/ (pA + pB)
    } else {
      stop(&quot;Invalid genotype match for 1 shared allele.&quot;)
    }
    
    LR &lt;- k0 + (k1 / Rxp)
    return(LR)
  }

  # Case 2: Two Shared Alleles
  if (shared_alleles == 2) {
    if (genotype_match == &quot;AA-AA&quot;) {
      Rxp &lt;- pA
      Rxu &lt;- pA^2
    } else if (genotype_match == &quot;AB-AB&quot;) {
      Rxp &lt;- (4 * pA * pB) / (pA + pB)
      Rxu &lt;- 2 * pA * pB
    } else {
      stop(&quot;Invalid genotype match for 2 shared alleles.&quot;)
    }

    LR &lt;- k0 + (k1 / Rxp) + (k2 / Rxu)
    return(LR)
  }
}</code></pre>
</div>
</div>
<div id="rscript-for-simulation" class="section level2">
<h2>Rscript for simulation</h2>
<p>Given our likelihood ratio calculation, we will now build a framework
around this for simulating pairs of individuals of known relationships
and calculating the likelihood ratio for each pair based on the
simulated genotypes.</p>
<p>The general flow of our simulation framework will be as follows:</p>
<p><img src="assets/simulation-framework.png" /></p>
<div id="step-1-simulation-setup" class="section level3">
<h3>Step 1: Simulation Setup</h3>
<p>The first step is to set up the simulation by defining the parameters
and functions required for generating the simulated data. This includes
specifying the populations, allele frequencies, and the kinship
coefficient matrix for likelihood ratio calculations.</p>
<pre class="r"><code>generate_simulation_setup &lt;- function(kinship_matrix, population_list, num_related, num_unrelated) {
  # Create an empty dataframe to store the simulation setup
  simulation_setup &lt;- data.frame(
    population = character(),
    relationship_type = character(),
    num_simulations = integer(),
    stringsAsFactors = FALSE
  )
  
  # Loop through each population and relationship type to create the setup
  for (population in population_list) {
    for (relationship in kinship_matrix$relationship_type) {
      num_simulations &lt;- ifelse(relationship == &quot;unrelated&quot;, num_unrelated, num_related)
      
      # Append to the simulation setup dataframe
      simulation_setup &lt;- rbind(simulation_setup, data.frame(
        population = population,
        relationship_type = relationship,
        num_simulations = num_simulations
      ))
    }
  }
  
  return(simulation_setup)
}</code></pre>
<div id="create-input-parameters" class="section level4">
<h4>Create input parameters</h4>
<p>The necessary parameters for this are initialized beforehand.</p>
<p>First, the kinship coefficients are provided in a matrix:</p>
<pre class="r"><code># Create a dataframe with relationship types and their respective kinship coefficients (k0, k1, k2)
kinship_matrix &lt;- tibble(
    relationship_type = factor(
      c(&quot;parent_child&quot;, &quot;full_siblings&quot;, &quot;half_siblings&quot;, &quot;cousins&quot;, &quot;second_cousins&quot;, &quot;unrelated&quot;),
      levels = c(&quot;parent_child&quot;, &quot;full_siblings&quot;, &quot;half_siblings&quot;, &quot;cousins&quot;, &quot;second_cousins&quot;, &quot;unrelated&quot;)
    ),
    k0 = c(0, 1/4, 1/2, 7/8, 15/16, 1),
    k1 = c(1, 1/2, 1/2, 1/8, 1/16, 0),
    k2 = c(0, 1/4, 0, 0, 0, 0)
  )

# Print the kinship matrix to check the contents
print(kinship_matrix)</code></pre>
<pre><code># A tibble: 6 × 4
  relationship_type    k0     k1    k2
  &lt;fct&gt;             &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;
1 parent_child      0     1       0   
2 full_siblings     0.25  0.5     0.25
3 half_siblings     0.5   0.5     0   
4 cousins           0.875 0.125   0   
5 second_cousins    0.938 0.0625  0   
6 unrelated         1     0       0   </code></pre>
<p>Then a list of populations is created:</p>
<pre><code># A tibble: 5 × 2
  population label           
  &lt;fct&gt;      &lt;chr&gt;           
1 all        All             
2 AfAm       African American
3 Cauc       Caucasian       
4 Hispanic   Hispanic        
5 Asian      Asian           </code></pre>
<pre><code>[1] &quot;all&quot;      &quot;AfAm&quot;     &quot;Cauc&quot;     &quot;Hispanic&quot; &quot;Asian&quot;   </code></pre>
</div>
<div id="gather-allele-frequency-information" class="section level4">
<h4>Gather allele frequency information</h4>
<pre><code>Classes &#39;data.table&#39; and &#39;data.frame&#39;:  14065 obs. of  4 variables:
 $ allele    : chr  &quot;2.2&quot; &quot;2.2&quot; &quot;2.2&quot; &quot;2.2&quot; ...
 $ marker    : chr  &quot;CSF1PO&quot; &quot;D10S1248&quot; &quot;D12S391&quot; &quot;D13S317&quot; ...
 $ frequency : num  0 0 0 0 0 0 0 0 0 0 ...
 $ population: chr  &quot;all&quot; &quot;all&quot; &quot;all&quot; &quot;all&quot; ...
 - attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; </code></pre>
<pre><code>   allele   marker frequency population
   &lt;char&gt;   &lt;char&gt;     &lt;num&gt;     &lt;char&gt;
1:    2.2   CSF1PO         0        all
2:    2.2 D10S1248         0        all
3:    2.2  D12S391         0        all
4:    2.2  D13S317         0        all
5:    2.2  D16S539         0        all
6:    2.2   D18S51         0        all</code></pre>
<p>We will now extract the unique loci from the allele frequency
tables:</p>
<pre><code> [1] &quot;CSF1PO&quot;   &quot;D10S1248&quot; &quot;D12S391&quot;  &quot;D13S317&quot;  &quot;D16S539&quot;  &quot;D18S51&quot;  
 [7] &quot;D19S433&quot;  &quot;D1S1656&quot;  &quot;D21S11&quot;   &quot;D22S1045&quot; &quot;D2S1338&quot;  &quot;D2S441&quot;  
[13] &quot;D3S1358&quot;  &quot;D5S818&quot;   &quot;D6S1043&quot;  &quot;D7S820&quot;   &quot;D8S1179&quot;  &quot;F13A01&quot;  
[19] &quot;F13B&quot;     &quot;FESFPS&quot;   &quot;FGA&quot;      &quot;LPL&quot;      &quot;Penta_C&quot;  &quot;Penta_D&quot; 
[25] &quot;Penta_E&quot;  &quot;SE33&quot;     &quot;TH01&quot;     &quot;TPOX&quot;     &quot;vWA&quot;     </code></pre>
<pre><code>Number of unique loci: 29 </code></pre>
<p>These are the 29 autosomal loci from the 2013 and 2017 FSI paper on
US STR allele frequencies for 29 autosomal STR loci <a
href="https://www.fsigenetics.com/article/S1872-4973(17)30180-1/fulltext">Steffen
et al 2017</a>.</p>
<p>We will use the list of different required loci to calculate
likelihood ratios for pairs of individuals. Below is a reference with
which loci are used in various sets.</p>
<pre><code>      locus core_13 identifiler_15 expanded_20 supplementary
1    CSF1PO       1              1           1             1
2       FGA       1              1           1             1
3      THO1       1              1           1             1
4      TPOX       1              1           1             1
5       vWA       1              1           1             1
6   D3S1358       1              1           1             1
7    D5S818       1              1           1             1
8    D7S820       1              1           1             1
9   D8S1179       1              1           1             1
10  D13S317       1              1           1             1
11  D16S539       1              1           1             1
12   D18S51       1              1           1             1
13   D21S11       1              1           1             1
14  D1S1656       0              0           1             1
15   D2S441       0              0           1             1
16  D2S1338       0              1           1             1
17 D10S1248       0              0           1             1
18  D12S391       0              0           1             1
19  D19S433       0              1           1             1
20 D22S1045       0              0           1             1
21     SE33       0              0           0             1
22  Penta_E       0              0           0             1
23  Penta_D       0              0           0             1</code></pre>
<pre><code>$core_13
 [1] &quot;CSF1PO&quot;  &quot;FGA&quot;     &quot;THO1&quot;    &quot;TPOX&quot;    &quot;vWA&quot;     &quot;D3S1358&quot; &quot;D5S818&quot; 
 [8] &quot;D7S820&quot;  &quot;D8S1179&quot; &quot;D13S317&quot; &quot;D16S539&quot; &quot;D18S51&quot;  &quot;D21S11&quot; 

$identifiler_15
 [1] &quot;CSF1PO&quot;  &quot;FGA&quot;     &quot;THO1&quot;    &quot;TPOX&quot;    &quot;vWA&quot;     &quot;D3S1358&quot; &quot;D5S818&quot; 
 [8] &quot;D7S820&quot;  &quot;D8S1179&quot; &quot;D13S317&quot; &quot;D16S539&quot; &quot;D18S51&quot;  &quot;D21S11&quot;  &quot;D2S1338&quot;
[15] &quot;D19S433&quot;

$expanded_20
 [1] &quot;CSF1PO&quot;   &quot;FGA&quot;      &quot;THO1&quot;     &quot;TPOX&quot;     &quot;vWA&quot;      &quot;D3S1358&quot; 
 [7] &quot;D5S818&quot;   &quot;D7S820&quot;   &quot;D8S1179&quot;  &quot;D13S317&quot;  &quot;D16S539&quot;  &quot;D18S51&quot;  
[13] &quot;D21S11&quot;   &quot;D1S1656&quot;  &quot;D2S441&quot;   &quot;D2S1338&quot;  &quot;D10S1248&quot; &quot;D12S391&quot; 
[19] &quot;D19S433&quot;  &quot;D22S1045&quot;

$supplementary
 [1] &quot;CSF1PO&quot;   &quot;FGA&quot;      &quot;THO1&quot;     &quot;TPOX&quot;     &quot;vWA&quot;      &quot;D3S1358&quot; 
 [7] &quot;D5S818&quot;   &quot;D7S820&quot;   &quot;D8S1179&quot;  &quot;D13S317&quot;  &quot;D16S539&quot;  &quot;D18S51&quot;  
[13] &quot;D21S11&quot;   &quot;D1S1656&quot;  &quot;D2S441&quot;   &quot;D2S1338&quot;  &quot;D10S1248&quot; &quot;D12S391&quot; 
[19] &quot;D19S433&quot;  &quot;D22S1045&quot; &quot;SE33&quot;     &quot;Penta_E&quot;  &quot;Penta_D&quot; 

$autosomal_29
 [1] &quot;CSF1PO&quot;   &quot;D10S1248&quot; &quot;D12S391&quot;  &quot;D13S317&quot;  &quot;D16S539&quot;  &quot;D18S51&quot;  
 [7] &quot;D19S433&quot;  &quot;D1S1656&quot;  &quot;D21S11&quot;   &quot;D22S1045&quot; &quot;D2S1338&quot;  &quot;D2S441&quot;  
[13] &quot;D3S1358&quot;  &quot;D5S818&quot;   &quot;D6S1043&quot;  &quot;D7S820&quot;   &quot;D8S1179&quot;  &quot;F13A01&quot;  
[19] &quot;F13B&quot;     &quot;FESFPS&quot;   &quot;FGA&quot;      &quot;LPL&quot;      &quot;Penta_C&quot;  &quot;Penta_D&quot; 
[25] &quot;Penta_E&quot;  &quot;SE33&quot;     &quot;TH01&quot;     &quot;TPOX&quot;     &quot;vWA&quot;     </code></pre>
<p>Now we can test the simulation setup function:</p>
<pre><code>   population relationship_type num_simulations
1         all      parent_child               5
2         all     full_siblings               5
3         all     half_siblings               5
4         all           cousins               5
5         all    second_cousins               5
6         all         unrelated              10
7        AfAm      parent_child               5
8        AfAm     full_siblings               5
9        AfAm     half_siblings               5
10       AfAm           cousins               5
11       AfAm    second_cousins               5
12       AfAm         unrelated              10
13       Cauc      parent_child               5
14       Cauc     full_siblings               5
15       Cauc     half_siblings               5
16       Cauc           cousins               5
17       Cauc    second_cousins               5
18       Cauc         unrelated              10
19   Hispanic      parent_child               5
20   Hispanic     full_siblings               5
21   Hispanic     half_siblings               5
22   Hispanic           cousins               5
23   Hispanic    second_cousins               5
24   Hispanic         unrelated              10
25      Asian      parent_child               5
26      Asian     full_siblings               5
27      Asian     half_siblings               5
28      Asian           cousins               5
29      Asian    second_cousins               5
30      Asian         unrelated              10</code></pre>
</div>
</div>
<div id="step-2-initialize-individuals" class="section level3">
<h3>Step 2: Initialize Individuals</h3>
<p>Once the general simulation setup is done, each pair of individuals
must be initialized as its own dataframe.</p>
<p>Now we can initialize the individuals:</p>
<div id="testing-the-initialization-function" class="section level4">
<h4>Testing the initialization function</h4>
<p>Here we test to see what the intialization function creates:</p>
<pre><code>    population relationship_type sim_id    locus ind1_allele1 ind1_allele2
        &lt;char&gt;            &lt;char&gt;  &lt;num&gt;   &lt;char&gt;       &lt;char&gt;       &lt;char&gt;
 1:        all      parent_child      1   CSF1PO                          
 2:        all      parent_child      1 D10S1248                          
 3:        all      parent_child      1  D12S391                          
 4:        all      parent_child      1  D13S317                          
 5:        all      parent_child      1  D16S539                          
 6:        all      parent_child      1   D18S51                          
 7:        all      parent_child      1  D19S433                          
 8:        all      parent_child      1  D1S1656                          
 9:        all      parent_child      1   D21S11                          
10:        all      parent_child      1 D22S1045                          
11:        all      parent_child      1  D2S1338                          
12:        all      parent_child      1   D2S441                          
13:        all      parent_child      1  D3S1358                          
14:        all      parent_child      1   D5S818                          
15:        all      parent_child      1  D6S1043                          
16:        all      parent_child      1   D7S820                          
17:        all      parent_child      1  D8S1179                          
18:        all      parent_child      1   F13A01                          
19:        all      parent_child      1     F13B                          
20:        all      parent_child      1   FESFPS                          
21:        all      parent_child      1      FGA                          
22:        all      parent_child      1      LPL                          
23:        all      parent_child      1  Penta_C                          
24:        all      parent_child      1  Penta_D                          
25:        all      parent_child      1  Penta_E                          
26:        all      parent_child      1     SE33                          
27:        all      parent_child      1     TH01                          
28:        all      parent_child      1     TPOX                          
29:        all      parent_child      1      vWA                          
    population relationship_type sim_id    locus ind1_allele1 ind1_allele2
    ind2_allele1 ind2_allele2 shared_alleles genotype_match    LR
          &lt;char&gt;       &lt;char&gt;          &lt;int&gt;         &lt;char&gt; &lt;num&gt;
 1:                                        0                    0
 2:                                        0                    0
 3:                                        0                    0
 4:                                        0                    0
 5:                                        0                    0
 6:                                        0                    0
 7:                                        0                    0
 8:                                        0                    0
 9:                                        0                    0
10:                                        0                    0
11:                                        0                    0
12:                                        0                    0
13:                                        0                    0
14:                                        0                    0
15:                                        0                    0
16:                                        0                    0
17:                                        0                    0
18:                                        0                    0
19:                                        0                    0
20:                                        0                    0
21:                                        0                    0
22:                                        0                    0
23:                                        0                    0
24:                                        0                    0
25:                                        0                    0
26:                                        0                    0
27:                                        0                    0
28:                                        0                    0
29:                                        0                    0
    ind2_allele1 ind2_allele2 shared_alleles genotype_match    LR</code></pre>
<pre><code>Unit: microseconds
       expr    min     lq     mean median     uq      max neval
 initialize 67.117 69.085 100.4115 71.545 75.112 2837.733   100</code></pre>
</div>
</div>
<div id="step-3-simulate-genotypes" class="section level3">
<h3>Step 3: Simulate Genotypes</h3>
<p>Next, we define a function to simulate the genotypes for each pair of
individuals based on the allele frequencies and kinship
coefficients.</p>
<pre><code>Unit: microseconds
     expr    min       lq     mean  median      uq      max neval
 geno_sim 548.99 569.1825 738.8274 593.311 619.633 11311.33   100</code></pre>
</div>
<div id="step-4-calculate-kinship" class="section level3">
<h3>Step 4: Calculate Kinship</h3>
<p>Then we define a related function to calculate the kinship and
likelihood ratios based on the simulated genotypes.</p>
<p>Here we test what the simulation will produce for each locus in a
pair of individuals.</p>
<pre><code>   population relationship_type sim_id  locus ind1_allele1 ind1_allele2
       &lt;char&gt;            &lt;char&gt;  &lt;num&gt; &lt;char&gt;       &lt;char&gt;       &lt;char&gt;
1:        all      parent_child      1 D21S11           29           30
   ind2_allele1 ind2_allele2 shared_alleles genotype_match    LR
         &lt;char&gt;       &lt;char&gt;          &lt;int&gt;         &lt;char&gt; &lt;num&gt;
1:           30           28              0                    0</code></pre>
<pre><code>   population relationship_type sim_id  locus ind1_allele1 ind1_allele2
       &lt;char&gt;            &lt;char&gt;  &lt;num&gt; &lt;char&gt;       &lt;char&gt;       &lt;char&gt;
1:        all      parent_child      1 D21S11           29           30
   ind2_allele1 ind2_allele2 shared_alleles genotype_match       LR
         &lt;char&gt;       &lt;char&gt;          &lt;int&gt;         &lt;char&gt;    &lt;num&gt;
1:           30           28              1          AB-AC 1.009747</code></pre>
<p>We then need a function to process the loci and calculate the kinship
for each row.</p>
<p>An example of the final output is shown below for one complete set of
loci for a pair of individuals:</p>
<pre><code>    population relationship_type sim_id    locus ind1_allele1 ind1_allele2
        &lt;char&gt;            &lt;char&gt;  &lt;num&gt;   &lt;char&gt;       &lt;char&gt;       &lt;char&gt;
 1:        all      parent_child      1   CSF1PO           11           10
 2:        all      parent_child      1 D10S1248           14           14
 3:        all      parent_child      1  D12S391         17.3           17
 4:        all      parent_child      1  D13S317           12           11
 5:        all      parent_child      1  D16S539            9           11
 6:        all      parent_child      1   D18S51           13           20
 7:        all      parent_child      1  D19S433           14         12.2
 8:        all      parent_child      1  D1S1656           14           16
 9:        all      parent_child      1   D21S11           30         30.2
10:        all      parent_child      1 D22S1045           15           16
11:        all      parent_child      1  D2S1338           19           21
12:        all      parent_child      1   D2S441           14           12
13:        all      parent_child      1  D3S1358           17           14
14:        all      parent_child      1   D5S818           13            9
15:        all      parent_child      1  D6S1043           12           12
16:        all      parent_child      1   D7S820           11            8
17:        all      parent_child      1  D8S1179           13           14
18:        all      parent_child      1   F13A01            7          3.2
19:        all      parent_child      1     F13B           10           10
20:        all      parent_child      1   FESFPS           10           12
21:        all      parent_child      1      FGA           21           20
22:        all      parent_child      1      LPL           12           10
23:        all      parent_child      1  Penta_C           12           11
24:        all      parent_child      1  Penta_D           10           11
25:        all      parent_child      1  Penta_E           12           16
26:        all      parent_child      1     SE33           16           16
27:        all      parent_child      1     TH01            9            9
28:        all      parent_child      1     TPOX            9           11
29:        all      parent_child      1      vWA           14           14
    population relationship_type sim_id    locus ind1_allele1 ind1_allele2
    ind2_allele1 ind2_allele2 shared_alleles genotype_match         LR
          &lt;char&gt;       &lt;char&gt;          &lt;int&gt;         &lt;char&gt;      &lt;num&gt;
 1:           10           13              1          AB-AC  1.0769231
 2:           15           14              1          AA-AB  1.6900489
 3:           17           19              1          AB-AC  2.0077519
 4:           13           12              1          AB-AC  0.8196203
 5:           11           11              1          AB-AA  1.7152318
 6:           13           19              1          AB-AC  2.3870968
 7:           13           14              1          AB-AC  0.8222222
 8:           14           14              1          AB-AA  3.4533333
 9:         30.2           32              1          AB-AC 11.5111111
10:           15           16              2          AB-AB  1.6198565
11:           19           19              1          AB-AA  3.3636364
12:           11           12              1          AB-AC  2.5145631
13:           16           14              1          AB-AC  2.8618785
14:           13            9              2          AB-AB  6.9283777
15:           12           11              1          AA-AB  2.3333333
16:            8           11              2          AB-AB  2.5760477
17:           14           10              1          AB-AC  1.0702479
18:            7          3.2              2          AB-AB  2.8123097
19:           10            8              1          AA-AB  1.4428969
20:           10           10              1          AB-AA  2.1949153
21:           21           20              2          AB-AB  4.5289617
22:           10           10              1          AB-AA  1.1562500
23:           12           12              1          AB-AA  2.3022222
24:            9           11              1          AB-AC  1.6071429
25:           16           17              1          AB-AC  4.9333333
26:         27.2           16              1          AA-AB 10.2574257
27:          9.3            9              1          AA-AB  2.9600000
28:            9           11              2          AB-AB  2.8385167
29:           18           14              1          AA-AB  5.2323232
    ind2_allele1 ind2_allele2 shared_alleles genotype_match         LR</code></pre>
</div>
<div id="step-5-process-simulation-setup" class="section level3">
<h3>Step 5: Process Simulation Setup</h3>
<p>Finally, we define a function to process the simulation setup and
initialize and process the pairs of individuals for each simulation
scenario.</p>
<pre><code>Unit: seconds
          expr      min       lq     mean   median      uq      max neval
 dplyr_version 2.570869 2.640026 2.730324 2.673396 2.72994 3.311632    10</code></pre>
</div>
<div id="plotting" class="section level3">
<h3>Plotting</h3>
<p><img src="figure/STR-simulation.Rmd/unnamed-chunk-20-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/f143ee1386425f8e06d3a94f09d512ac794e407f/docs/figure/STR-simulation.Rmd/unnamed-chunk-20-1.png" target="_blank">f143ee1</a>
</td>
<td>
tinalasisi
</td>
<td>
2024-09-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/b98be9778d2e820b4bc587f3596d68fa71e5a53b/docs/figure/STR-simulation.Rmd/unnamed-chunk-20-1.png" target="_blank">b98be97</a>
</td>
<td>
linmatch
</td>
<td>
2024-08-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>`summarise()` has grouped output by &#39;relationship_type&#39;, &#39;population&#39;. You can
override using the `.groups` argument.</code></pre>
<p><img src="figure/STR-simulation.Rmd/unnamed-chunk-22-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/f143ee1386425f8e06d3a94f09d512ac794e407f/docs/figure/STR-simulation.Rmd/unnamed-chunk-22-1.png" target="_blank">f143ee1</a>
</td>
<td>
tinalasisi
</td>
<td>
2024-09-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/b98be9778d2e820b4bc587f3596d68fa71e5a53b/docs/figure/STR-simulation.Rmd/unnamed-chunk-22-1.png" target="_blank">b98be97</a>
</td>
<td>
linmatch
</td>
<td>
2024-08-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># A tibble: 25 × 7
   population loci_set  fixed_cutoff cutoff_1 cutoff_0_1 cutoff_0_01 n_unrelated
   &lt;fct&gt;      &lt;fct&gt;            &lt;dbl&gt;    &lt;dbl&gt;      &lt;dbl&gt;       &lt;dbl&gt;       &lt;int&gt;
 1 all        core_13              1        1          1           1          10
 2 all        identifi…            1        1          1           1          10
 3 all        expanded…            1        1          1           1          10
 4 all        suppleme…            1        1          1           1          10
 5 all        autosoma…            1        1          1           1          10
 6 AfAm       core_13              1        1          1           1          10
 7 AfAm       identifi…            1        1          1           1          10
 8 AfAm       expanded…            1        1          1           1          10
 9 AfAm       suppleme…            1        1          1           1          10
10 AfAm       autosoma…            1        1          1           1          10
# ℹ 15 more rows</code></pre>
<pre><code># A tibble: 625 × 8
   population relationship_type loci_set       proportion_exceeding_fixed
   &lt;fct&gt;      &lt;fct&gt;             &lt;fct&gt;                               &lt;dbl&gt;
 1 all        parent_child      core_13                                 1
 2 all        parent_child      core_13                                 1
 3 all        parent_child      core_13                                 1
 4 all        parent_child      core_13                                 1
 5 all        parent_child      core_13                                 1
 6 all        parent_child      identifiler_15                          1
 7 all        parent_child      identifiler_15                          1
 8 all        parent_child      identifiler_15                          1
 9 all        parent_child      identifiler_15                          1
10 all        parent_child      identifiler_15                          1
# ℹ 615 more rows
# ℹ 4 more variables: proportion_exceeding_1 &lt;dbl&gt;,
#   proportion_exceeding_0_1 &lt;dbl&gt;, proportion_exceeding_0_01 &lt;dbl&gt;,
#   n_related &lt;int&gt;</code></pre>
<p><img src="figure/STR-simulation.Rmd/unnamed-chunk-25-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/f143ee1386425f8e06d3a94f09d512ac794e407f/docs/figure/STR-simulation.Rmd/unnamed-chunk-25-1.png" target="_blank">f143ee1</a>
</td>
<td>
tinalasisi
</td>
<td>
2024-09-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/b98be9778d2e820b4bc587f3596d68fa71e5a53b/docs/figure/STR-simulation.Rmd/unnamed-chunk-25-1.png" target="_blank">b98be97</a>
</td>
<td>
linmatch
</td>
<td>
2024-08-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="simulation-results" class="section level2">
<h2>Simulation results</h2>
<pre><code>Rows: 360000 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (3): population, known_relationship_type, tested_relationship_type
dbl (3): replicate_id, num_shared_alleles_sum, log_R_sum

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div
id="proportion-of-individuals-of-known-relationship-type-exceeding-likelihood-cut-off"
class="section level2">
<h2>Proportion of individuals of known relationship type exceeding
likelihood cut-off</h2>
<pre><code>  population relationship_type fp_rate prop_exceeding
1       AfAm      parent_child     0.1          1.000
2      Asian      parent_child     0.1          1.000
3       Cauc      parent_child     0.1          1.000
4   Hispanic      parent_child     0.1          1.000
5       AfAm     full_siblings     0.1          0.794
6      Asian     full_siblings     0.1          0.820</code></pre>
<pre><code>`summarise()` has grouped output by &#39;population&#39;. You can override using the
`.groups` argument.</code></pre>
<p><img src="figure/STR-simulation.Rmd/unnamed-chunk-29-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/b98be9778d2e820b4bc587f3596d68fa71e5a53b/docs/figure/STR-simulation.Rmd/unnamed-chunk-29-1.png" target="_blank">b98be97</a>
</td>
<td>
linmatch
</td>
<td>
2024-08-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cut-offs-for-each-fpr" class="section level2">
<h2>Cut-offs for each FPR</h2>
<pre><code>`summarise()` has grouped output by &#39;population&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code># A tibble: 24 × 7
   population known_relationship_type mean_log_R_sum median_log_R_sum
   &lt;chr&gt;      &lt;chr&gt;                            &lt;dbl&gt;            &lt;dbl&gt;
 1 AfAm       cousins                         -0.533            2.42 
 2 AfAm       full_siblings                   10.5             11.5  
 3 AfAm       half_siblings                    3.85             6.80 
 4 AfAm       parent_child                    19.6             19.0  
 5 AfAm       second_cousins                  -1.23             1.78 
 6 AfAm       unrelated                       -2.00             0.914
 7 Asian      cousins                         -0.538            2.42 
 8 Asian      full_siblings                   10.5             11.6  
 9 Asian      half_siblings                    3.82             6.78 
10 Asian      parent_child                    19.5             18.9  
# ℹ 14 more rows
# ℹ 3 more variables: min_log_R_sum &lt;dbl&gt;, max_log_R_sum &lt;dbl&gt;, count &lt;int&gt;</code></pre>
<div
id="calculate-llr-cutoffs-for-unrelated-pairs-at-each-fpr-by-population"
class="section level4">
<h4>Calculate LLR cutoffs for unrelated pairs at each FPR by
population</h4>
<p>This table presents aggregated log_R_sum values by relationship type
for each population, alongside LLR cutoffs calculated for unrelated
pairs to maintain false positive rates (FPRs) of 1%, 0.1%, and 0.01%.
The cutoffs indicate the LLR threshold above which a pair is less likely
to be unrelated at the given FPR, thereby serving as a critical
benchmark for assessing relationship evidence. The mean log_R_sum values
further elucidate the average strength of genetic evidence supporting
each relationship type within populations, highlighting variations and
consistencies in genetic relatedness indicators across demographic
groups.</p>
<table class="table table-striped table-hover table-condensed" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
population
</th>
<th style="text-align:right;">
cousins
</th>
<th style="text-align:right;">
full_siblings
</th>
<th style="text-align:right;">
half_siblings
</th>
<th style="text-align:right;">
parent_child
</th>
<th style="text-align:right;">
second_cousins
</th>
<th style="text-align:right;">
unrelated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
AfAm
</td>
<td style="text-align:right;">
-0.5334790
</td>
<td style="text-align:right;">
10.45417
</td>
<td style="text-align:right;">
3.849582
</td>
<td style="text-align:right;">
19.62211
</td>
<td style="text-align:right;">
-1.229432
</td>
<td style="text-align:right;">
-2.003701
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Asian
</td>
<td style="text-align:right;">
-0.5379709
</td>
<td style="text-align:right;">
10.45853
</td>
<td style="text-align:right;">
3.821000
</td>
<td style="text-align:right;">
19.54267
</td>
<td style="text-align:right;">
-1.387000
</td>
<td style="text-align:right;">
-1.982983
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Cauc
</td>
<td style="text-align:right;">
-0.5075911
</td>
<td style="text-align:right;">
10.48275
</td>
<td style="text-align:right;">
3.873924
</td>
<td style="text-align:right;">
19.51039
</td>
<td style="text-align:right;">
-1.218428
</td>
<td style="text-align:right;">
-1.982207
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Hispanic
</td>
<td style="text-align:right;">
-0.5536143
</td>
<td style="text-align:right;">
10.40972
</td>
<td style="text-align:right;">
3.837237
</td>
<td style="text-align:right;">
19.58039
</td>
<td style="text-align:right;">
-1.249979
</td>
<td style="text-align:right;">
-1.984905
</td>
</tr>
</tbody>
</table>
<p><img src="figure/STR-simulation.Rmd/unnamed-chunk-37-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/b98be9778d2e820b4bc587f3596d68fa71e5a53b/docs/figure/STR-simulation.Rmd/unnamed-chunk-37-1.png" target="_blank">b98be97</a>
</td>
<td>
linmatch
</td>
<td>
2024-08-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/STR-simulation.Rmd/unnamed-chunk-39-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/b98be9778d2e820b4bc587f3596d68fa71e5a53b/docs/figure/STR-simulation.Rmd/unnamed-chunk-39-1.png" target="_blank">b98be97</a>
</td>
<td>
linmatch
</td>
<td>
2024-08-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Detroit
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.16.4    dtplyr_1.3.1         furrr_0.3.1         
 [4] future_1.34.0        kableExtra_1.4.0     microbenchmark_1.5.0
 [7] patchwork_1.3.0      lubridate_1.9.4      forcats_1.0.0       
[10] stringr_1.5.1        dplyr_1.1.4          purrr_1.0.4         
[13] readr_2.1.5          tidyr_1.3.1          tibble_3.2.1        
[16] ggplot2_3.5.1        tidyverse_2.0.0      RColorBrewer_1.1-3  
[19] wesanderson_0.3.7    workflowr_1.7.1     

loaded via a namespace (and not attached):
 [1] gtable_0.3.6      xfun_0.50         bslib_0.9.0       processx_3.8.5   
 [5] callr_3.7.6       tzdb_0.4.0        vctrs_0.6.5       tools_4.4.2      
 [9] ps_1.8.1          generics_0.1.3    parallel_4.4.2    pkgconfig_2.0.3  
[13] lifecycle_1.0.4   compiler_4.4.2    farver_2.1.2      git2r_0.35.0     
[17] textshaping_1.0.0 munsell_0.5.1     codetools_0.2-20  getPass_0.2-4    
[21] httpuv_1.6.15     htmltools_0.5.8.1 sass_0.4.9        yaml_2.3.10      
[25] crayon_1.5.3      later_1.4.1       pillar_1.10.1     jquerylib_0.1.4  
[29] whisker_0.4.1     cachem_1.1.0      parallelly_1.42.0 tidyselect_1.2.1 
[33] digest_0.6.37     stringi_1.8.4     listenv_0.9.1     labeling_0.4.3   
[37] rprojroot_2.0.4   fastmap_1.2.0     grid_4.4.2        colorspace_2.1-1 
[41] cli_3.6.3         magrittr_2.0.3    utf8_1.2.4        withr_3.0.2      
[45] scales_1.3.0      promises_1.3.2    bit64_4.6.0-1     timechange_0.3.0 
[49] rmarkdown_2.29    httr_1.4.7        globals_0.16.3    bit_4.5.0.1      
[53] ragg_1.3.3        hms_1.1.3         evaluate_1.0.3    knitr_1.49       
[57] viridisLite_0.4.2 rlang_1.1.5       Rcpp_1.0.14       glue_1.8.0       
[61] xml2_1.3.6        vroom_1.6.5       svglite_2.1.3     rstudioapi_0.17.1
[65] jsonlite_1.8.9    R6_2.5.1          systemfonts_1.2.1 fs_1.6.5         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
